{
 "cells": [
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2026-02-08T08:56:32.995060Z",
     "start_time": "2026-02-08T08:56:30.035326Z"
    }
   },
   "source": [
    "\n",
    "import fhirclient.models.binary as bin\n",
    "import json\n",
    "from pypdf import PdfReader\n",
    "import pymupdf\n",
    "import tabula\n",
    "import base64\n",
    "import io\n",
    "import pymupdf4llm\n",
    "import pathlib\n",
    "\n",
    "report_list = ['ORU_R01_DLIMS.txt','ORU_R01_R125.1_R0A.txt','ORU_R01_R125.1_RBS.txt','ORU_R01_R125.1_REP.txt','ORU_R01_R125.1_RR8.txt','ORU_R01_R125.1_RX1.txt','ORU_R01_R125.1_SG9.txt','ORU_R01_R125.1_ZT001.txt','ORU_R01_R125.1_7A3.txt','ORU_R01_R125.1_RPY.txt','ORU_R01_R125.1_RXK.txt', 'SHIRE_ORU_R01_RM3.txt', 'WALES_ORU_R01_TX.txt']\n",
    "\n",
    "# For testing\n",
    "\n",
    "report_list = ['ORU_R01_R125.1_R0A.txt','ORU_R01_DLIMS.txt']\n",
    "\n",
    "\n",
    "pdfReaderEnabled = False\n",
    "pdfPymupdfEnabled = False\n",
    "pdfpymupdf4llmEnabled = True\n",
    "pdfTableEnabled = False\n",
    "\n",
    "for file in report_list:\n",
    "    with open('Output/FHIR/R01/' + file+ '.json', 'rb') as g:\n",
    "        rawJson = g.read()\n",
    "        jsonBundle = json.loads(rawJson)\n",
    "\n",
    "        #bundle = b.Bundle(jsonBundle)\n",
    "        for entry in jsonBundle['entry']:\n",
    "            #print(entry['resource']['resourceType'])\n",
    "            if entry['resource']['resourceType'] == 'Binary':\n",
    "                binary = bin.Binary(entry['resource'])\n",
    "                encoded = binary.data\n",
    "                #print(encoded)\n",
    "                decode = base64.b64decode(encoded)\n",
    "                pdf_stream = io.BytesIO(decode)\n",
    "                if pdfpymupdf4llmEnabled:\n",
    "                    doc = pymupdf.open(\"pdf\", pdf_stream)\n",
    "                    md_text = pymupdf4llm.to_markdown(doc)\n",
    "                    print(md_text)\n",
    "                    pathlib.Path('Output/Markdown/R01/' + file+ '.md').write_bytes(md_text.encode())\n",
    "                if pdfPymupdfEnabled:\n",
    "                    # https://pymupdf.readthedocs.io/en/latest/the-basics.html\n",
    "\n",
    "                    # Initial Tests indicate this is the best method to use.\n",
    "\n",
    "                    doc = pymupdf.open(\"pdf\", pdf_stream)\n",
    "                    print('Pages: '+ str(len(doc)))\n",
    "                    print('============ pymupdf ' + file + ' ==============')\n",
    "                    for page in doc: # iterate the document pages\n",
    "                        text = page.get_text().encode(\"utf8\")\n",
    "                        print(text)\n",
    "                if pdfReaderEnabled:\n",
    "                    pdf = PdfReader(pdf_stream)\n",
    "                    print('Pages: '+ str(len(pdf.pages)))\n",
    "                    print('============ PDF Reader ' + file + ' ==============')\n",
    "                    for page in pdf.pages:\n",
    "                        print(page.extract_text())\n",
    "                if pdfTableEnabled:\n",
    "                    tables = tabula.read_pdf(pdf_stream\n",
    "                                             , pages='all', multiple_tables=True)\n",
    "                    print('============ Tabula ' + file + ' ==============')\n",
    "                    for table in tables:\n",
    "                        print(table)\n",
    "                with open('Output/PDF/R01/' + file+ '.pdf', 'wb') as pdf:\n",
    "                    pdf.write(decode)\n",
    "\n",
    "\n",
    "pdf_list = ['R24-0WH7-1.pdf','R24-1A4W-1.pdf']\n",
    "\n",
    "for file in pdf_list:\n",
    "    with open('Output/PDF/R01/' + file, 'rb') as g:\n",
    "        pdf_stream = g.read()\n",
    "        if pdfpymupdf4llmEnabled:\n",
    "            doc = pymupdf.open(\"pdf\", pdf_stream)\n",
    "            md_text = pymupdf4llm.to_markdown(doc)\n",
    "            print(md_text)\n",
    "            pathlib.Path('Output/Markdown/R01/' + file+ '.md').write_bytes(md_text.encode())\n",
    "\n",
    "\n"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dr EQA\n",
      "Obstetrics & Gynaecology\n",
      "GenQA Trust\n",
      "Address\n",
      "\n",
      "\n",
      "\n",
      "**Genomic Diagnostics Laboratory (GDL)**\n",
      "\n",
      "Manchester Centre for Genomic Medicine (MCGM)\n",
      "\n",
      "6th Floor, Saint Mary's Hospital, Oxford Road\n",
      "\n",
      "Manchester, M13 9WL, United Kingdom\n",
      "\n",
      "\n",
      "E-mail: mft.genomics@nhs.net\n",
      "\n",
      "Tel: 0161 276 6122/ 6553\n",
      "\n",
      "www.mangen.co.uk\n",
      "\n",
      "\n",
      "# **No Sample Received TEST: Karyotype Analysis Report**\n",
      "\n",
      "\n",
      "\n",
      "Forename: **Fetus of Lotte** NHS Number:\n",
      "\n",
      "\n",
      "\n",
      "Surname: **Schmidt**\n",
      "\n",
      "\n",
      "\n",
      "GDL Referral No: **R24-1A4W**\n",
      "\n",
      "\n",
      "Date of Referral: **10/09/2024**\n",
      "\n",
      "\n",
      "\n",
      "DoB: **13/08/1995** Sex: **Unknown** Date of Report:\n",
      "\n",
      "\n",
      "\n",
      "**13/08/1995**\n",
      "\n",
      "\n",
      "\n",
      "Hospital No: **2024RK44**\n",
      "\n",
      "\n",
      "Sex: **Unknown**\n",
      "\n",
      "\n",
      "\n",
      "Referral Reason: **IUGR, clinodactyly. Follow up to an array showing a female fetus with loss of 15q26.1 to 15qter.**\n",
      "\n",
      "**Date sample taken: 09/09/2024**\n",
      "\n",
      "\n",
      "ISCN: **46,XX,r(15)(::p11q26.1::)**\n",
      "\n",
      "\n",
      "**G-BANDED KARYOTYPE RESULT:**\n",
      "Chromosome analysis of cultured amniotic fluid cells from this patient has shown an abnormal unbalanced female\n",
      "karyotype in which one chromosome 15 is present as a ring chromosome with a terminal deletion of the long arm from\n",
      "band q26.1. This is consistent with the loss detected by array studies which do not provide positional information about\n",
      "chromosome structure.\n",
      "\n",
      "\n",
      "Ring chromosome 15 with a terminal deletion of the long arm including the IGF1R gene [OMIM *147370, chromosome\n",
      "15q26.3 locus] are associated with a broad spectrum of congenital abnormalities commonly including pre and post-natal\n",
      "growth retardation and clinodactyly. [1,2,3 ] Any loss of material from the short arm of chromosome 15 in the formation of\n",
      "the ring is likely to be non-coding heterochromatin and will not have any additional phenotypic effect. This finding,\n",
      "therefore, is considered to be causative of the fetal abnormalities noted on scan.\n",
      "\n",
      "\n",
      "Ring chromosomes can be unstable mitotically and this result does not exclude the possibility of chromosome 15\n",
      "mosaicism in other tissues.\n",
      "\n",
      "\n",
      "Should this pregnancy progress to term, please ensure that a copy of this report is placed in the child's notes at delivery.\n",
      "\n",
      "\n",
      "**GENETICS URGENCY REFERRAL:**\n",
      "We recommend that Lotte and her partner are referred to their local Genetics Urgency Clinic where this report and its\n",
      "implications can be discussed in more detail.\n",
      "\n",
      "\n",
      "**PARENTAL TESTING:**\n",
      "In the majority of cases, a ring chromosome 15 arises as a new event with a low parental risk of recurrence. In rare cases\n",
      "they can be inherited from a parent who carries a rearrangement predisposing to this finding. At an appropriate time,\n",
      "please send blood samples from Lotte and her partner [5ml in lithium heparin quoting the laboratory reference R241A4W] to your local genetics laboratory for karyotype analysis to evaluate the origin and parental recurrence risk. Please\n",
      "provide relevant parental phenotypic information to assist in the reporting of test findings.\n",
      "\n",
      "\n",
      "**References:**\n",
      "1. Unique leaflet \"Ring 15\"  https://rarechromo.org/media/information/Chromosome%2015/Ring%2015%20FTNW.pdf\n",
      "2. Unique leaflet \"15q26 deletions\" https://rarechromo.org/media/information/Chromosome%2015/15q26%\n",
      "20deletions%20FTNW.pdf\n",
      "3. Traisrisilp, K.et al Prenatal Sonographic Features of Ring Chromosome 15: A Case Report and Literature Review.\n",
      "Diagnostics 2022, 12, 885. https://doi.o\n",
      "\n",
      "\n",
      "**Methodology and Technical Information:**\n",
      "Cells have been cultured and harvested using standard operating procedures. Chromosome analysis of the complete set\n",
      "of GTG-banded metaphase chromosomes has been undertaken on a minimum of 3 cells at a minimum resolution level of\n",
      "300-bands, in accordance with ACGS best practice guidelines. Cells were digitally captured, and chromosomes paired,\n",
      "using the Neon analysis software. Subtle or cryptic structural rearrangements/imbalances may not have been detected\n",
      "and this result may not reflect the karyotype of cells in other tissues. Cytogenetic heteromorphisms that reflect normal\n",
      "chromosome variation are not reported.\n",
      "\n",
      "\n",
      "Page 1 of 2\n",
      "\n",
      "\n",
      "Forename: **Fetus of Lotte** NHS Number: GDL Referral No: **R24-1A4W**\n",
      "\n",
      "Surname: **Schmidt** Hospital No: **2024RK44** Date of Referral: **10/09/2024**\n",
      "DoB: **13/08/1995** Date of Report:\n",
      "\n",
      "\n",
      "\n",
      "NHS Number:\n",
      "Hospital No:\n",
      "\n",
      "\n",
      "\n",
      "**2024RK44**\n",
      "\n",
      "\n",
      "\n",
      "**Regulation and accreditations:**\n",
      "All aspects of this test are within the accredited scope _ANONYMISED._\n",
      "\n",
      "\n",
      "**Notes to assessors:**\n",
      "**1. No indication on referral form that QF-PCR studies have been performed. If they had been we would reference them**\n",
      "**in the main body of the report.**\n",
      "\n",
      "\n",
      "\n",
      "Analysed by:\n",
      "Edward McHale\n",
      "Clinical Scientist\n",
      "\n",
      "\n",
      "Indication:\n",
      "Test Type:\n",
      "GenU Score:\n",
      "GenU Banding:\n",
      "\n",
      "\n",
      "\n",
      "Checked by:\n",
      "Steve Trueman\n",
      "Clinical Scientist\n",
      "\n",
      "\n",
      "\n",
      "Authorised by:\n",
      "\n",
      "\n",
      "\n",
      "NEQAS\n",
      "TEST: Karyotype\n",
      "\n",
      "\n",
      "\n",
      "Page 2 of 2\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "execution_count": 12
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
