{
 "cells": [
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2026-02-09T06:03:31.134551Z",
     "start_time": "2026-02-09T06:03:20.377492Z"
    }
   },
   "source": [
    "\n",
    "import fhirclient.models.binary as bin\n",
    "import json\n",
    "import spacy\n",
    "from spacy import displacy\n",
    "from spacypdfreader.spacypdfreader import pdf_reader\n",
    "from spacy import displacy\n",
    "from pypdf import PdfReader\n",
    "import pymupdf\n",
    "import tabula\n",
    "import base64\n",
    "import io\n",
    "import pymupdf4llm\n",
    "import pathlib\n",
    "\n",
    "report_list = ['ORU_R01_DLIMS.txt','ORU_R01_R125.1_R0A.txt','ORU_R01_R125.1_RBS.txt','ORU_R01_R125.1_REP.txt','ORU_R01_R125.1_RR8.txt','ORU_R01_R125.1_RX1.txt','ORU_R01_R125.1_SG9.txt','ORU_R01_R125.1_ZT001.txt','ORU_R01_R125.1_7A3.txt','ORU_R01_R125.1_RPY.txt','ORU_R01_R125.1_RXK.txt', 'SHIRE_ORU_R01_RM3.txt', 'WALES_ORU_R01_TX.txt']\n",
    "\n",
    "# For testing\n",
    "\n",
    "report_list = ['ORU_R01_R125.1_R0A.txt','ORU_R01_DLIMS.txt']\n",
    "\n",
    "\n",
    "pdfReaderEnabled = False\n",
    "pdfPymupdfEnabled = False\n",
    "pdfpymupdf4llmEnabled = True\n",
    "pdfTableEnabled = False\n",
    "\n",
    "\n",
    "for file in report_list:\n",
    "    with open('Output/FHIR/R01/' + file+ '.json', 'rb') as g:\n",
    "        rawJson = g.read()\n",
    "        jsonBundle = json.loads(rawJson)\n",
    "\n",
    "        #bundle = b.Bundle(jsonBundle)\n",
    "        for entry in jsonBundle['entry']:\n",
    "            #print(entry['resource']['resourceType'])\n",
    "            if entry['resource']['resourceType'] == 'Binary':\n",
    "                binary = bin.Binary(entry['resource'])\n",
    "                encoded = binary.data\n",
    "                #print(encoded)\n",
    "                decode = base64.b64decode(encoded)\n",
    "                pdf_stream = io.BytesIO(decode)\n",
    "                if pdfpymupdf4llmEnabled:\n",
    "                    doc = pymupdf.open(\"pdf\", pdf_stream)\n",
    "                    md_text = pymupdf4llm.to_markdown(doc)\n",
    "                    #print(md_text)\n",
    "                    pathlib.Path('Output/Markdown/R01/' + file+ '.md').write_bytes(md_text.encode())\n",
    "                if pdfPymupdfEnabled:\n",
    "                    # https://pymupdf.readthedocs.io/en/latest/the-basics.html\n",
    "\n",
    "                    # Initial Tests indicate this is the best method to use.\n",
    "\n",
    "                    doc = pymupdf.open(\"pdf\", pdf_stream)\n",
    "                    print('Pages: '+ str(len(doc)))\n",
    "                    print('============ pymupdf ' + file + ' ==============')\n",
    "                    for page in doc: # iterate the document pages\n",
    "                        text = page.get_text().encode(\"utf8\")\n",
    "                        print(text)\n",
    "                if pdfReaderEnabled:\n",
    "                    pdf = PdfReader(pdf_stream)\n",
    "                    print('Pages: '+ str(len(pdf.pages)))\n",
    "                    print('============ PDF Reader ' + file + ' ==============')\n",
    "                    for page in pdf.pages:\n",
    "                        text = page.extract_text()\n",
    "                        # print(text)\n",
    "                if pdfTableEnabled:\n",
    "                    tables = tabula.read_pdf(pdf_stream\n",
    "                                             , pages='all', multiple_tables=True)\n",
    "                    print('============ Tabula ' + file + ' ==============')\n",
    "                    for table in tables:\n",
    "                        tab1 = table\n",
    "                        #print(table)\n",
    "                with open('Output/PDF/R01/' + file+ '.pdf', 'wb') as pdf:\n",
    "                    pdf.write(decode)\n",
    "\n",
    "\n",
    "pdf_list = ['R24-0WH7-1.pdf','R24-1A4W-1.pdf']\n",
    "\n",
    "for file in pdf_list:\n",
    "    with open('Output/PDF/R01/' + file, 'rb') as g:\n",
    "        pdf_stream = g.read()\n",
    "        if pdfpymupdf4llmEnabled:\n",
    "            doc = pymupdf.open(\"pdf\", pdf_stream)\n",
    "            md_text = pymupdf4llm.to_markdown(doc)\n",
    "            print(md_text)\n",
    "            pathlib.Path('Output/Markdown/R01/' + file+ '.md').write_bytes(md_text.encode())\n",
    "\n",
    "\n",
    "pdf_list = ['genomic-medicine-device-test-order-form-cancer-v1.22.pdf']\n",
    "\n",
    "for file in pdf_list:\n",
    "    with open('Output/PDF/O21/' + file, 'rb') as g:\n",
    "        pdf_stream = g.read()\n",
    "        if pdfpymupdf4llmEnabled:\n",
    "            doc = pymupdf.open(\"pdf\", pdf_stream)\n",
    "            md_text = pymupdf4llm.to_markdown(doc)\n",
    "            print(md_text)\n",
    "            pathlib.Path('Output/Markdown/O21/' + file+ '.md').write_bytes(md_text.encode())\n",
    "\n"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NHS Genomic Medicine Service, WGS Test Request Cancer, July 2024 v1.22 to be used for WGS go-live.\n",
      "This document is subject to version control and is regularly updated. Please confirm you are using the current version by\n",
      "contacting your local Genomic Laboratory Hub\n",
      "\n",
      "~~**Genomic Medicine Service**~~\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "|Whole Genome Sequencing (WGS) Test Request PLEASE DO NOT USE FOR NON-WGS TESTS CANCER|Col2|\n",
      "|---|---|\n",
      "|**Requesting organisation:**<br> <br>|**Requesting organisation:**<br> <br>|\n",
      "|<br>~~**GLH laboratory to receive sample:**~~<br>North West|~~Test Required~~<br>**Whole Genome Sequencing **|\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "|GLH laboratory to receive sample: North West|Col2|Col3|Col4|Test Required Whole Genome Sequencing|Col6|Col7|\n",
      "|---|---|---|---|---|---|---|\n",
      "|~~Patient first name~~<br> <br>Myrcella|~~Patient first name~~<br> <br>Myrcella|~~Patient first name~~<br> <br>Myrcella|~~Patient first name~~<br> <br>Myrcella|~~Ethnicity~~<br>|~~Ethnicity~~<br>|~~Ethnicity~~<br>|\n",
      "|~~Patient last name~~<br> <br> <br>Wrexham|~~Patient last name~~<br> <br> <br>Wrexham|~~Patient last name~~<br> <br> <br>Wrexham|~~Patient last name~~<br> <br> <br>Wrexham|~~Test Directory Clinical Indication & code (cancer type &~~<br>~~sub-type)~~<br>**_The clinical indications listed at the bottom of the pick_**<br>**_list under 'NEW INDICATIONS' are not live for all NHS_**<br>**_GLHs. Please check with GLHs prior to ordering._**|~~Test Directory Clinical Indication & code (cancer type &~~<br>~~sub-type)~~<br>**_The clinical indications listed at the bottom of the pick_**<br>**_list under 'NEW INDICATIONS' are not live for all NHS_**<br>**_GLHs. Please check with GLHs prior to ordering._**|~~Test Directory Clinical Indication & code (cancer type &~~<br>~~sub-type)~~<br>**_The clinical indications listed at the bottom of the pick_**<br>**_list under 'NEW INDICATIONS' are not live for all NHS_**<br>**_GLHs. Please check with GLHs prior to ordering._**|\n",
      "|~~Date of birth~~_(dd/mm/yyyy)_ <br><br>16/10/1970|~~Date of birth~~_(dd/mm/yyyy)_ <br><br>16/10/1970|~~Hospital number~~<br>12345678|~~Hospital number~~<br>12345678|~~Hospital number~~<br>12345678|~~Hospital number~~<br>12345678|~~Hospital number~~<br>12345678|\n",
      "|~~Gender~~<br> Male<br> Female<br> Other<br><br>|~~Gender~~<br> Male<br> Female<br> Other<br><br>|~~Gender~~<br> Male<br> Female<br> Other<br><br>|~~Gender~~<br> Male<br> Female<br> Other<br><br>|Presentation status<br> First diagnosis<br> Recurrence / Relapse<br> Unknown <br>|Presentation status<br> First diagnosis<br> Recurrence / Relapse<br> Unknown <br>|Presentation status<br> First diagnosis<br> Recurrence / Relapse<br> Unknown <br>|\n",
      "|Postcode<br> <br>N<br>G<br>1<br>0<br>1<br>n<br>p|Postcode<br> <br>N<br>G<br>1<br>0<br>1<br>n<br>p|Postcode<br> <br>N<br>G<br>1<br>0<br>1<br>n<br>p|Postcode<br> <br>N<br>G<br>1<br>0<br>1<br>n<br>p|~~Additional clinical information (if required)~~<br>_E.g. previous tumours, molecular testing, and relevant treatment history with_<br>_date(s)_<br>Some clinical information|~~Additional clinical information (if required)~~<br>_E.g. previous tumours, molecular testing, and relevant treatment history with_<br>_date(s)_<br>Some clinical information|~~Additional clinical information (if required)~~<br>_E.g. previous tumours, molecular testing, and relevant treatment history with_<br>_date(s)_<br>Some clinical information|\n",
      "|~~NHS number~~<br>~~Reason NHS Number not availab~~le:<br> Patient not eligible for NHS number (e.g. foreign national)<br> Other (provide reason):<br>|~~NHS number~~<br>~~Reason NHS Number not availab~~le:<br> Patient not eligible for NHS number (e.g. foreign national)<br> Other (provide reason):<br>|~~NHS number~~<br>~~Reason NHS Number not availab~~le:<br> Patient not eligible for NHS number (e.g. foreign national)<br> Other (provide reason):<br>|~~NHS number~~<br>~~Reason NHS Number not availab~~le:<br> Patient not eligible for NHS number (e.g. foreign national)<br> Other (provide reason):<br>|~~NHS number~~<br>~~Reason NHS Number not availab~~le:<br> Patient not eligible for NHS number (e.g. foreign national)<br> Other (provide reason):<br>|~~NHS number~~<br>~~Reason NHS Number not availab~~le:<br> Patient not eligible for NHS number (e.g. foreign national)<br> Other (provide reason):<br>|~~NHS number~~<br>~~Reason NHS Number not availab~~le:<br> Patient not eligible for NHS number (e.g. foreign national)<br> Other (provide reason):<br>|\n",
      "|~~**Solid tumour requests only**~~<br> <br> <br>|~~**Solid tumour requests only**~~<br> <br> <br>|~~**Solid tumour requests only**~~<br> <br> <br>|~~**Solid tumour requests only**~~<br> <br> <br>|~~**Solid tumour requests only**~~<br> <br> <br>|~~**Solid tumour requests only**~~<br> <br> <br>|~~**Solid tumour requests only**~~<br> <br> <br>|\n",
      "|~~Primary~~<br> Metastatic<br> Unknown<br> Lymphoma<br> <br>\u0014|~~Histopathology Lab ID~~<br> <br>1234567PATH|~~Histopathology Lab ID~~<br> <br>1234567PATH|~~Histopathology Lab ID~~<br> <br>1234567PATH|~~Additional tumour information (if relevant)~~<br>_E.g. site of metastasis (if metastatic), or unknown primary_<br> <br>|~~Additional tumour information (if relevant)~~<br>_E.g. site of metastasis (if metastatic), or unknown primary_<br> <br>|~~Additional tumour information (if relevant)~~<br>_E.g. site of metastasis (if metastatic), or unknown primary_<br> <br>|\n",
      "|~~Primary~~<br> Metastatic<br> Unknown<br> Lymphoma<br> <br>\u0014|~~Date of this diagnosis~~_(dd/mm/yyyy)_ <br>|~~Date of this diagnosis~~_(dd/mm/yyyy)_ <br>|~~Date of this diagnosis~~_(dd/mm/yyyy)_ <br>|~~Tumour topography ~~|~~Tumour morphology ~~|~~Tumour morphology ~~|\n",
      "|~~**Haemato-oncology liquid tumour requests only**~~<br> <br> <br> <br> <br>|~~**Haemato-oncology liquid tumour requests only**~~<br> <br> <br> <br> <br>|~~**Haemato-oncology liquid tumour requests only**~~<br> <br> <br> <br> <br>|~~**Haemato-oncology liquid tumour requests only**~~<br> <br> <br> <br> <br>|~~**Haemato-oncology liquid tumour requests only**~~<br> <br> <br> <br> <br>|~~**Haemato-oncology liquid tumour requests only**~~<br> <br> <br> <br> <br>|~~**Haemato-oncology liquid tumour requests only**~~<br> <br> <br> <br> <br>|\n",
      "|~~ AML~~<br>~~ ALL~~<br>~~ Other (please specify):~~<br>|~~ AML~~<br>~~ ALL~~<br>~~ Other (please specify):~~<br>|~~ AML~~<br>~~ ALL~~<br>~~ Other (please specify):~~<br>|~~ AML~~<br>~~ ALL~~<br>~~ Other (please specify):~~<br>|~~ SIHMDS Lab ID~~<br>|~~Date of this diagnosis~~_(dd/mm/yyyy)_|~~Date of this diagnosis~~_(dd/mm/yyyy)_|\n",
      "|~~**Complete for tumour samples**~~~~_(being sent to GLH DNA extraction lab)_~~ <br> <br> <br> <br>|~~**Complete for tumour samples**~~~~_(being sent to GLH DNA extraction lab)_~~ <br> <br> <br> <br>|~~**Complete for tumour samples**~~~~_(being sent to GLH DNA extraction lab)_~~ <br> <br> <br> <br>|~~**Complete for tumour samples**~~~~_(being sent to GLH DNA extraction lab)_~~ <br> <br> <br> <br>|~~**Complete for tumour samples**~~~~_(being sent to GLH DNA extraction lab)_~~ <br> <br> <br> <br>|~~**Complete for tumour samples**~~~~_(being sent to GLH DNA extraction lab)_~~ <br> <br> <br> <br>|~~**Complete for tumour samples**~~~~_(being sent to GLH DNA extraction lab)_~~ <br> <br> <br> <br>|\n",
      "|~~ Fresh frozen tumour~~<br>~~ Bone marrow~~<br>~~ Blood (EDTA)~~<br>~~ Other (please specify):~~<br> <br>\u0014|~~ Fresh frozen tumour~~<br>~~ Bone marrow~~<br>~~ Blood (EDTA)~~<br>~~ Other (please specify):~~<br> <br>\u0014|~~ Fresh frozen tumour~~<br>~~ Bone marrow~~<br>~~ Blood (EDTA)~~<br>~~ Other (please specify):~~<br> <br>\u0014|~~ Fresh frozen tumour~~<br>~~ Bone marrow~~<br>~~ Blood (EDTA)~~<br>~~ Other (please specify):~~<br> <br>\u0014|~~ Fresh frozen tumour~~<br>~~ Bone marrow~~<br>~~ Blood (EDTA)~~<br>~~ Other (please specify):~~<br> <br>\u0014|~~ Fresh frozen tumour~~<br>~~ Bone marrow~~<br>~~ Blood (EDTA)~~<br>~~ Other (please specify):~~<br> <br>\u0014|~~ Fresh frozen tumour~~<br>~~ Bone marrow~~<br>~~ Blood (EDTA)~~<br>~~ Other (please specify):~~<br> <br>\u0014|\n",
      "|<br> <br> <br> <br>~~_% malignant nuclei / blasts or equivalent in this sample (refer to sample handling guidance) must be provided below_~~<br> <br> <br> <br>|<br> <br> <br> <br>~~_% malignant nuclei / blasts or equivalent in this sample (refer to sample handling guidance) must be provided below_~~<br> <br> <br> <br>|<br> <br> <br> <br>~~_% malignant nuclei / blasts or equivalent in this sample (refer to sample handling guidance) must be provided below_~~<br> <br> <br> <br>|<br> <br> <br> <br>~~_% malignant nuclei / blasts or equivalent in this sample (refer to sample handling guidance) must be provided below_~~<br> <br> <br> <br>|<br> <br> <br> <br>~~_% malignant nuclei / blasts or equivalent in this sample (refer to sample handling guidance) must be provided below_~~<br> <br> <br> <br>|<br> <br> <br> <br>~~_% malignant nuclei / blasts or equivalent in this sample (refer to sample handling guidance) must be provided below_~~<br> <br> <br> <br>|<br> <br> <br> <br>~~_% malignant nuclei / blasts or equivalent in this sample (refer to sample handling guidance) must be provided below_~~<br> <br> <br> <br>|\n",
      "|~~ Sample ID~~|~~ Sample ID~~|~~ Sample ID~~|~~Collection date / time~~|~~% Malignant nuclei / blasts~~|~~% Malignant nuclei / blasts~~|~~If BM/PB provide volume~~<br>and nucleated cell count|\n",
      "|<br>SAMPLEID1|<br>SAMPLEID1|<br>SAMPLEID1|||||\n",
      "|~~**Complete for germline samples**~~~~_(being sent to GLH DNA extraction lab)_ ~~<br> <br> <br> <br> <br>|~~**Complete for germline samples**~~~~_(being sent to GLH DNA extraction lab)_ ~~<br> <br> <br> <br> <br>|~~**Complete for germline samples**~~~~_(being sent to GLH DNA extraction lab)_ ~~<br> <br> <br> <br> <br>|~~**Complete for germline samples**~~~~_(being sent to GLH DNA extraction lab)_ ~~<br> <br> <br> <br> <br>|~~**Complete for germline samples**~~~~_(being sent to GLH DNA extraction lab)_ ~~<br> <br> <br> <br> <br>|~~**Complete for germline samples**~~~~_(being sent to GLH DNA extraction lab)_ ~~<br> <br> <br> <br> <br>|~~**Complete for germline samples**~~~~_(being sent to GLH DNA extraction lab)_ ~~<br> <br> <br> <br> <br>|\n",
      "|~~ Blood (EDTA)~~<br>~~ Saliva~~<br>~~ Fibroblasts~~<br>~~ Skin biopsy~~<br>~~ Other (please specify):~~<br> <br> <br> <br>|~~ Blood (EDTA)~~<br>~~ Saliva~~<br>~~ Fibroblasts~~<br>~~ Skin biopsy~~<br>~~ Other (please specify):~~<br> <br> <br> <br>|~~ Blood (EDTA)~~<br>~~ Saliva~~<br>~~ Fibroblasts~~<br>~~ Skin biopsy~~<br>~~ Other (please specify):~~<br> <br> <br> <br>|~~ Blood (EDTA)~~<br>~~ Saliva~~<br>~~ Fibroblasts~~<br>~~ Skin biopsy~~<br>~~ Other (please specify):~~<br> <br> <br> <br>|~~ Blood (EDTA)~~<br>~~ Saliva~~<br>~~ Fibroblasts~~<br>~~ Skin biopsy~~<br>~~ Other (please specify):~~<br> <br> <br> <br>|~~ Blood (EDTA)~~<br>~~ Saliva~~<br>~~ Fibroblasts~~<br>~~ Skin biopsy~~<br>~~ Other (please specify):~~<br> <br> <br> <br>|~~ Blood (EDTA)~~<br>~~ Saliva~~<br>~~ Fibroblasts~~<br>~~ Skin biopsy~~<br>~~ Other (please specify):~~<br> <br> <br> <br>|\n",
      "|~~Sample ID~~|~~Sample ID~~|~~Sample ID~~|~~Collection date / time~~|~~Sample volume if applicable~~|~~Sample volume if applicable~~|~~Comments~~|\n",
      "||||||||\n",
      "|~~**Responsible consultant **~~|~~**Responsible consultant **~~|~~**Responsible consultant **~~|~~**Responsible consultant **~~|<br>~~**Main contact (if different from responsible consultant)**~~|<br>~~**Main contact (if different from responsible consultant)**~~|<br>~~**Main contact (if different from responsible consultant)**~~|\n",
      "|<br>Name:<br>Department address:<br>Phone:<br>Email:|<br>Name:<br>Department address:<br>Phone:<br>Email:|<br>Name:<br>Department address:<br>Phone:<br>Email:|<br>Name:<br>Department address:<br>Phone:<br>Email:|~~Name:~~<br>Department address:<br>Phone:<br>Email:<br>|~~Name:~~<br>Department address:<br>Phone:<br>Email:<br>|~~Name:~~<br>Department address:<br>Phone:<br>Email:<br>|\n",
      "\n",
      "\n",
      "Patient conversation taken place; Record of Discussion form to follow\n",
      "\n",
      "\n",
      "Version 1.22 Page 1 of 1\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "execution_count": 3
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "## Preparing Textual Data for Statistics and Machine Learning\n",
    "\n",
    "### Tokens"
   ],
   "id": "3d4bfbf14579d2a"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-02-09T06:03:32.392623Z",
     "start_time": "2026-02-09T06:03:31.135887Z"
    }
   },
   "cell_type": "code",
   "source": [
    "\n",
    "\n",
    "file = 'R24-0WH7-1.pdf'\n",
    "spacyEnabled = True\n",
    "\n",
    "if spacyEnabled:\n",
    "    print('============ Spacy ' + file + ' ==============')\n",
    "    nlp = spacy.load('en_core_web_sm')\n",
    "    #nlp = spacy.load('en_core_web_trf')\n",
    "    # TODO use pdf_stream instead\n",
    "    doc = pdf_reader('Output/PDF/R01/' + file, nlp)\n",
    "\n",
    "    for token in doc:\n",
    "        print(token,end=\"|\")"
   ],
   "id": "c82b5bb90642d50d",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "North|West|Genomic|Laboratory|Hub|(|Manchester|Site|)|\n",
      "|Manchester|Centre|for|Genomic|Medicine|\n",
      "|6th|Floor|,|St|Mary|'s|Hospital|,|Manchester|,|M13|9WL|\n",
      "|Scientific|Operational|Director|:|Dr|E.|Howard|\n",
      "|https://mft.nhs.uk/nwglh/| |mft.genomics@nhs.net| |Tel|+44(0|)|161|276|6122|\n",
      "\n",
      "|Forename|:|\n",
      "|Surname|:|\n",
      "|Sex|:|\n",
      "|DoB|:|\n",
      "|NHS|No|:|\n",
      "\n",
      "|ACCESSIBILITY|\n",
      "|ZZZTESTA|\n",
      "|Unspecified|\n",
      "|21/10/1978|\n",
      "|999|001|2563|\n",
      "\n",
      "|Report|No|:|\n",
      "|R24|-|0WH7|-|1|\n",
      "|Hospital|No|:|Not|provided|\n",
      "|Not|provided|\n",
      "|Other|No|:|\n",
      "|Blood|\n",
      "|Sample|:|\n",
      "\n",
      "|Sample2|:|\n",
      "\n",
      "|Blood|\n",
      "\n",
      "|Collected|:|\n",
      "|Collected2|:|\n",
      "\n",
      "|Not|provided|\n",
      "|25/06/2024|\n",
      "\n",
      "|Received|:|\n",
      "|Received2|:|\n",
      "\n",
      "|08/10/2023|08:18|\n",
      "|26/06/2024|14:21|\n",
      "\n",
      "|Activated|:|\n",
      "|Reported|:|\n",
      "\n",
      "|26/06/2024|\n",
      "\n",
      "|Referred|by|:|Dr|Anna|Castleton|,|Haematology|,|The|Christie|Hospital|,|Manchester|,|M20|4BX|,|(|chn-tr.Haematology.Results@nhs.net|)|\n",
      "\n",
      "|Genomics|Laboratory|Report|\n",
      "\n",
      "|Reason|for|Testing|\n",
      "|MRD|follow|up|.|\n",
      "\n",
      "|RESULT|SUMMARY|:|\n",
      "\n",
      "|BCR::ABL1|Major|(|e14a2|/|e13a2|)|–|Detected|and|quantitated|.|See|graph|overleaf|.|\n",
      "\n",
      "|Result|and|Interpretation|\n",
      "|The|bone|marrow|/|peripheral|blood|sample|taken|/|received|from|this|patient|on|XX|/|XX|/|XX|showed|detectable|BCR::ABL1|transcripts|:|X%|\n",
      "|BCR::ABL1IS|(|MRX|)|.|\n",
      "\n",
      "|[|REASON|FOR|TESTING|-|COPY|BELOW|TEXT|TO|'|REPORTING|REFERRAL|INFORMATION|TEXT|'|BOX|THEN|DELETE|]|\n",
      "|Measurable|residual|disease|analysis|for|the|BCR::ABL1|fusion|transcript|has|been|requested|.|\n",
      "\n",
      "|Reported|By|:|\n",
      "|Hannah|Reed|\n",
      "|Pre|-|Registration|Scientist|\n",
      "\n",
      "|Authorised|By|:|\n",
      "\n",
      "|Page|1|of|2|\n",
      "\n",
      "\f|Forename|:|\n",
      "|Surname|:|\n",
      "\n",
      "|ACCESSIBILITY|\n",
      "|ZZZTESTA|\n",
      "\n",
      "|APPENDIX|I|-|Test|Methodology|:|\n",
      "\n",
      "|Report|No|:|\n",
      "|DoB|:|\n",
      "\n",
      "|R24|-|0WH7|-|1|\n",
      "|21/10/1978|\n",
      "\n",
      "|Interpretation|of|graph|:|BCR::ABL1|RT|-|qPCR|results|for|the|major|transcripts|(|e14a2|/|e13a2|)|are|expressed|on|the|International|Scale|(|IS|)|as|a|percentage|relative|to|the|\n",
      "|standardised|baseline|used|in|the|IRIS|trial|which|evaluated|the|tyrosine|kinase|inhibitor|(|TKI|)|in|patients|with|CML|.|The|reference|gene|used|for|this|assay|is|ABL1|.|Levels|\n",
      "|above|the|upper|relapse|risk|threshold|are|consistent|with|fusion|gene|levels|detected|at|presentation|,|persistent|disease|and|relapse|.|The|lower|risk|threshold|(|or|lower|)|\n",
      "|approximately|equates|to|a|2|-|log|reduction|which|is|usually|consistent|with|complete|cytogenetic|remission|(|CCyR|)|.|The|pale|blue|line|(|or|lower|)|represents|a|major|\n",
      "|molecular|response|(|MMR|)|defined|as|a|3|-|log|reduction|from|the|standardised|baseline|(|MR3|or|0.1|%|BCR::ABL1|IS|)|in|major|(|e14a2|/|e13a2|)|positive|patients|.|For|major|\n",
      "|(|e14a2|/|e13a2|)|positive|patients|,|deep|molecular|responses|(|DMR|)|of|MR4|,|MR4.5|,|and|MR5|are|defined|as|≤0.01|%|,|≤0.0032|%|and|≤0.001|%|BCR::ABL1|IS|respectively|.|\n",
      "|Please|note|,|these|levels|have|been|defined|for|CML|patients|,|equivalence|has|not|been|established|for|ALL|patients|.|The|levels|are|not|patient|specific|and|therefore|there|\n",
      "|may|be|some|patient|outliers|.|The|target|sensitivity|of|the|assay|for|major|(|e14a2|/|e13a2|)|positive|patients|is|MR4.5|in|most|samples|.|Levels|in|-|between|the|two|risk|\n",
      "|thresholds|(|red|lines|)|are|consistent|with|gradual|decrease|in|disease|level|after|treatment|(|As|after|diagnosis|or|relapse|)|ot|increased|/|high|risk|of|relapse|if|the|patients|in|\n",
      "|remission|.|A|level|of|0.0001|%|is|negative|which|represents|undetectable|transcripts|.|Red|crosses|on|graph|indicate|unquantifiable|results|.|Green|data|points|show|the|\n",
      "|sensitivity|level|of|each|test|.|Sensitivity|levels|of|0.00001|%|and|below|are|consistent|with|highest|reliability|.|*|For|optimum|sensitivity|,|please|provide|at|least|5ml|EDTA|\n",
      "|peripheral|blood|or|1ml|EDTA|bone|marrow|within|24|hours|(|preferable|)|to|48|hours|.|Interpretation|of|BCR::ABL1|major|(|e14a2|/|e13a2|)|results|will|be|according|to|the|current|\n",
      "|ELN|guidelines|(|Baccarani|et|al|2013|,|Blood|,|122|,|872|-|884|)|,|depending|on|sufficient|clinical|information|given|.|\n",
      "\n",
      "|Laboratory|process|\n",
      "|RNA|extraction|:|Maxwell|RSC|simplyRNA|for|samples|extracted|by|NWGLH|\n",
      "|Testing|method|:|The|TRUPCR|®|BCR|-|ABL1|Kit|(|Version|2.1|)|by|3B|BlackBio|Biotech|India|Ltd|(|operating|as|TRUPCR|®|Europe|Ltd|)|is|used|to|quantify|BCR::ABL1|Major|\n",
      "|(|e14a2|and|e13a2|)|fusion|transcripts|in|RNA|isolated|from|the|patient|sample|.|Reverse|transcription|quantitative|polymerase|chain|reaction|(|RT|-|qPCR|)|is|performed|on|the|\n",
      "|Applied|Biosystems|®|QuantStudio|™|6|platform|and|analysed|in|the|accompanying|software|.|The|TRUPCR|®|BCR|-|ABL1|Kit|includes|six|plasmid|-|based|standards|that|are|\n",
      "|calibrated|to|European|reference|material|(|ERM|®|-AD623|)|.|Each|standard|contains|both|BCR::ABL1|and|ABL1|targets|to|limit|variability|,|with|copy|numbers|of|10|^|2|-|10|^|7|for|\n",
      "|standard|curve|production|.|Each|kit|contains|a|calibrator|sample|which|has|been|normalised|to|WHO|International|Standard|NIBSC|09/138|material|for|quantitation|of|BCR|:|:|\n",
      "|ABL1|,|enabling|internal|conversion|of|results|to|the|International|Scale|(|IS|)|.|\n",
      "\n",
      "|Limitations|:|The|assay|is|designed|for|measuring|deep|molecular|response|and|has|a|limit|of|detection|of|≥3|copies|of|BCR::ABL1|transcript|per|reaction|well|.|\n",
      "\n",
      "|Regulation|and|accreditations|:|This|test|is|not|UKAS|accredited|.|This|test|has|been|validated|locally|and|will|be|submitted|for|UKAS|assessment|imminently|.|\n",
      "\n",
      "|Please|contact|the|laboratory|for|additional|details|,|if|required|.|\n",
      "\n",
      "|APPENDIX|II|-|Minimal|/|Measurable|Residual|Disease|(|MRD|)|Chart|:|\n",
      "\n",
      "|Page|2|of|2|\n",
      "\n",
      "\f|"
     ]
    }
   ],
   "execution_count": 4
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### Extracting Named Entities",
   "id": "a14597d207d473fe"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-02-09T06:03:32.460018Z",
     "start_time": "2026-02-09T06:03:32.436856Z"
    }
   },
   "cell_type": "code",
   "source": [
    " if spacyEnabled:\n",
    "     for ent in doc.ents:\n",
    "        print(ent.text, ent.label_)"
   ],
   "id": "88640e4c07362bb1",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "North West Genomic Laboratory Hub ORG\n",
      "Manchester Site PERSON\n",
      "Manchester Centre PERSON\n",
      "6th Floor ORG\n",
      "St Mary's PERSON\n",
      "Manchester GPE\n",
      "M13 PERSON\n",
      "Scientific Operational ORG\n",
      "E. Howard PERSON\n",
      "161 CARDINAL\n",
      "276 6122 DATE\n",
      "999 CARDINAL\n",
      "001 2563\n",
      "\n",
      "Report DATE\n",
      "Anna Castleton PERSON\n",
      "Haematology ORG\n",
      "The Christie Hospital ORG\n",
      "Manchester PERSON\n",
      "M20 PERSON\n",
      "Major (e14a2 ORG\n",
      "XX GPE\n",
      "XX GPE\n",
      "MRX ORG\n",
      "Hannah Reed\n",
      "Pre-Registration Scientist PERSON\n",
      "1 CARDINAL\n",
      "2 CARDINAL\n",
      "the International Scale (IS ORG\n",
      "IRIS ORG\n",
      "TKI ORG\n",
      "CML ORG\n",
      "ABL1 PERSON\n",
      "2 CARDINAL\n",
      "MMR ORG\n",
      "3 CARDINAL\n",
      "MR3 GPE\n",
      "0.1% PERCENT\n",
      "CML ORG\n",
      "two CARDINAL\n",
      "0.0001% PERCENT\n",
      "0.00001% PERCENT\n",
      "at least 5ml CARDINAL\n",
      "1ml ORDINAL\n",
      "24 hours TIME\n",
      "48 hours TIME\n",
      "Baccarani PERSON\n",
      "Blood GPE\n",
      "122 CARDINAL\n",
      "872 CARDINAL\n",
      "Maxwell RSC simplyRNA PERSON\n",
      "NWGLH ORG\n",
      "2.1 CARDINAL\n",
      "3B CARDINAL\n",
      "BlackBio Biotech India Ltd ORG\n",
      "TRUPCR® Europe Ltd ORG\n",
      "RNA ORG\n",
      "Applied Biosystems ORG\n",
      "QuantStudio PERSON\n",
      "6 CARDINAL\n",
      "six CARDINAL\n",
      "European NORP\n",
      "ABL1 PERSON\n",
      "10 CARDINAL\n",
      "7 CARDINAL\n",
      "WHO International Standard ORG\n",
      "ABL1 PERSON\n",
      "the International Scale ORG\n",
      "UKAS NORP\n",
      "UKAS NORP\n",
      "MRD ORG\n",
      "2 CARDINAL\n",
      "2 CARDINAL\n"
     ]
    }
   ],
   "execution_count": 5
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-02-09T06:03:32.519966Z",
     "start_time": "2026-02-09T06:03:32.462746Z"
    }
   },
   "cell_type": "code",
   "source": [
    "if spacyEnabled:\n",
    "    displacy.render(doc, style='ent')"
   ],
   "id": "15689d79a00ec9bf",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ],
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">\n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    North West Genomic Laboratory Hub\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       " (\n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    Manchester Site\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       ")<br>\n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    Manchester Centre\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       " for Genomic Medicine<br>\n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    6th Floor\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       ", \n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    St Mary's\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       " Hospital, \n",
       "<mark class=\"entity\" style=\"background: #feca74; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    Manchester\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">GPE</span>\n",
       "</mark>\n",
       ", \n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    M13\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       " 9WL<br>\n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    Scientific Operational\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       " Director: Dr \n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    E. Howard\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       "<br>https://mft.nhs.uk/nwglh/  mft.genomics@nhs.net  Tel +44(0) \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    161\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       " \n",
       "<mark class=\"entity\" style=\"background: #bfe1d9; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    276 6122\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">DATE</span>\n",
       "</mark>\n",
       "<br><br>Forename:<br>Surname:<br>Sex:<br>DoB:<br>NHS No:<br><br>ACCESSIBILITY<br>ZZZTESTA<br>Unspecified<br>21/10/1978<br>\n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    999\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       " \n",
       "<mark class=\"entity\" style=\"background: #bfe1d9; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    001 2563\n",
       "\n",
       "Report\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">DATE</span>\n",
       "</mark>\n",
       " No:<br>R24-0WH7-1<br>Hospital No: Not provided<br>Not provided<br>Other No:<br>Blood<br>Sample:<br><br>Sample2:<br><br>Blood<br><br>Collected:<br>Collected2:<br><br>Not provided<br>25/06/2024<br><br>Received:<br>Received2:<br><br>08/10/2023 08:18<br>26/06/2024 14:21<br><br>Activated:<br>Reported:<br><br>26/06/2024<br><br>Referred by: Dr \n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    Anna Castleton\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       ", \n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    Haematology\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       ", \n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    The Christie Hospital\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       ", \n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    Manchester\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       ", \n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    M20\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       " 4BX, (chn-tr.Haematology.Results@nhs.net)<br><br>Genomics Laboratory Report<br><br>Reason for Testing<br>MRD follow up.<br><br>RESULT SUMMARY:<br><br>BCR::ABL1 \n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    Major (e14a2\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       " / e13a2) – Detected and quantitated. See graph overleaf.<br><br>Result and Interpretation<br>The bone marrow/peripheral blood sample taken/received from this patient on \n",
       "<mark class=\"entity\" style=\"background: #feca74; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    XX\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">GPE</span>\n",
       "</mark>\n",
       "/\n",
       "<mark class=\"entity\" style=\"background: #feca74; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    XX\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">GPE</span>\n",
       "</mark>\n",
       "/XX showed detectable BCR::ABL1 transcripts: X%<br>BCR::ABL1IS (\n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    MRX\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       ").<br><br>[REASON FOR TESTING - COPY BELOW TEXT TO 'REPORTING REFERRAL INFORMATION TEXT' BOX THEN DELETE]<br>Measurable residual disease analysis for the BCR::ABL1 fusion transcript has been requested.<br><br>Reported By:<br>\n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    Hannah Reed\n",
       "Pre-Registration Scientist\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       "<br><br>Authorised By:<br><br>Page \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    1\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       " of \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    2\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       "<br><br>\fForename:<br>Surname:<br><br>ACCESSIBILITY<br>ZZZTESTA<br><br>APPENDIX I - Test Methodology:<br><br>Report No:<br>DoB:<br><br>R24-0WH7-1<br>21/10/1978<br><br>Interpretation of graph: BCR::ABL1 RT-qPCR results for the major transcripts (e14a2/e13a2) are expressed on \n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    the International Scale (IS\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       ") as a percentage relative to the<br>standardised baseline used in the \n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    IRIS\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       " trial which evaluated the tyrosine kinase inhibitor (\n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    TKI\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       ") in patients with \n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    CML\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       ". The reference gene used for this assay is \n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ABL1\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       ". Levels<br>above the upper relapse risk threshold are consistent with fusion gene levels detected at presentation, persistent disease and relapse. The lower risk threshold (or lower)<br>approximately equates to a \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    2\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       "-log reduction which is usually consistent with complete cytogenetic remission (CCyR). The pale blue line (or lower) represents a major<br>molecular response (\n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    MMR\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       ") defined as a \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    3\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       "-log reduction from the standardised baseline (\n",
       "<mark class=\"entity\" style=\"background: #feca74; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    MR3\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">GPE</span>\n",
       "</mark>\n",
       " or \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    0.1%\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\n",
       "</mark>\n",
       " BCR::ABL1 IS) in major (e14a2/e13a2) positive patients. For major<br>(e14a2/e13a2) positive patients, deep molecular responses (DMR) of MR4, MR4.5, and MR5 are defined as ≤0.01%, ≤0.0032% and ≤0.001% BCR::ABL1 IS respectively.<br>Please note, these levels have been defined for \n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    CML\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       " patients, equivalence has not been established for ALL patients. The levels are not patient specific and therefore there<br>may be some patient outliers. The target sensitivity of the assay for major (e14a2/e13a2) positive patients is MR4.5 in most samples. Levels in-between the \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    two\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       " risk<br>thresholds (red lines) are consistent with gradual decrease in disease level after treatment (As after diagnosis or relapse) ot increased/high risk of relapse if the patients in<br>remission. A level of \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    0.0001%\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\n",
       "</mark>\n",
       " is negative which represents undetectable transcripts. Red crosses on graph indicate unquantifiable results. Green data points show the<br>sensitivity level of each test. Sensitivity levels of \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    0.00001%\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\n",
       "</mark>\n",
       " and below are consistent with highest reliability. * For optimum sensitivity, please provide \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    at least 5ml\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       " EDTA<br>peripheral blood or \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    1ml\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORDINAL</span>\n",
       "</mark>\n",
       " EDTA bone marrow within \n",
       "<mark class=\"entity\" style=\"background: #bfe1d9; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    24 hours\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">TIME</span>\n",
       "</mark>\n",
       " (preferable) to \n",
       "<mark class=\"entity\" style=\"background: #bfe1d9; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    48 hours\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">TIME</span>\n",
       "</mark>\n",
       ". Interpretation of BCR::ABL1 major (e14a2/e13a2) results will be according to the current<br>ELN guidelines (\n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    Baccarani\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       " et al 2013, \n",
       "<mark class=\"entity\" style=\"background: #feca74; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    Blood\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">GPE</span>\n",
       "</mark>\n",
       ", \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    122\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       ", \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    872\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       "-884), depending on sufficient clinical information given. <br><br>Laboratory process<br>RNA extraction: \n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    Maxwell RSC simplyRNA\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       " for samples extracted by \n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    NWGLH\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       "<br>Testing method: The TRUPCR® BCR-ABL1 Kit (Version \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    2.1\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       ") by \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    3B\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       " \n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    BlackBio Biotech India Ltd\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       " (operating as \n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    TRUPCR® Europe Ltd\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       ") is used to quantify BCR::ABL1 Major<br>(e14a2 and e13a2) fusion transcripts in \n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    RNA\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       " isolated from the patient sample. Reverse transcription quantitative polymerase chain reaction (RT-qPCR) is performed on the<br>\n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    Applied Biosystems\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       "® \n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    QuantStudio\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       "™ \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    6\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       " platform and analysed in the accompanying software. The TRUPCR® BCR-ABL1 Kit includes \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    six\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       " plasmid-based standards that are<br>calibrated to \n",
       "<mark class=\"entity\" style=\"background: #c887fb; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    European\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">NORP</span>\n",
       "</mark>\n",
       " reference material (ERM®-AD623). Each standard contains both BCR::ABL1 and \n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ABL1\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       " targets to limit variability, with copy numbers of \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    10\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       "^2-10^\n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    7\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       " for<br>standard curve production. Each kit contains a calibrator sample which has been normalised to \n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    WHO International Standard\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       " NIBSC 09/138 material for quantitation of BCR::<br>\n",
       "<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ABL1\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\n",
       "</mark>\n",
       ", enabling internal conversion of results to \n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    the International Scale\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       " (IS).<br><br>Limitations: The assay is designed for measuring deep molecular response and has a limit of detection of ≥3 copies of BCR::ABL1 transcript per reaction well.<br><br>Regulation and accreditations: This test is not \n",
       "<mark class=\"entity\" style=\"background: #c887fb; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    UKAS\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">NORP</span>\n",
       "</mark>\n",
       " accredited. This test has been validated locally and will be submitted for \n",
       "<mark class=\"entity\" style=\"background: #c887fb; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    UKAS\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">NORP</span>\n",
       "</mark>\n",
       " assessment imminently.<br><br>Please contact the laboratory for additional details, if required.<br><br>APPENDIX II - Minimal/Measurable Residual Disease (\n",
       "<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    MRD\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\n",
       "</mark>\n",
       ") Chart:<br><br>Page \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    2\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       " of \n",
       "<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    2\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\n",
       "</mark>\n",
       "<br><br>\f</div></span>"
      ]
     },
     "metadata": {},
     "output_type": "display_data",
     "jetTransient": {
      "display_id": null
     }
    }
   ],
   "execution_count": 6
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### medspacy",
   "id": "149a85a96b0ec842"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-02-09T06:08:46.388845Z",
     "start_time": "2026-02-09T06:08:45.074233Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import scispacy\n",
    "from scispacy.abbreviation import AbbreviationDetector\n",
    "\n",
    "nlp = spacy.load(\"en_core_sci_sm\")\n",
    "\n",
    "nlp.add_pipe(\"abbreviation_detector\")\n",
    "\n",
    "doc = nlp(md_text)\n",
    "\n",
    "print(\"Abbreviation\", \"\\t\", \"Definition\")\n",
    "for abrv in doc._.abbreviations:\n",
    "    print(f\"{abrv} \\t ({abrv.start}, {abrv.end}) {abrv._.long_form}\")\n"
   ],
   "id": "6027790bac2b23c",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Abbreviation \t Definition\n",
      "WGS \t (58, 59) |Whole Genome Sequencing\n",
      "WGS \t (5, 6) |Whole Genome Sequencing\n",
      "WGS \t (17, 18) |Whole Genome Sequencing\n",
      "sample \t (1402, 1403) sample handling guidance\n",
      "sample \t (1326, 1327) sample handling guidance\n",
      "sample \t (1398, 1399) sample handling guidance\n",
      "sample \t (1322, 1323) sample handling guidance\n",
      "sample \t (102, 103) sample handling guidance\n",
      "sample \t (1550, 1551) sample handling guidance\n",
      "sample \t (1364, 1365) sample handling guidance\n",
      "sample \t (1440, 1441) sample handling guidance\n",
      "sample \t (1478, 1479) sample handling guidance\n",
      "sample \t (1516, 1517) sample handling guidance\n",
      "sample \t (1512, 1513) sample handling guidance\n",
      "sample \t (1554, 1555) sample handling guidance\n",
      "sample \t (1436, 1437) sample handling guidance\n",
      "sample \t (1474, 1475) sample handling guidance\n",
      "sample \t (1360, 1361) sample handling guidance\n"
     ]
    }
   ],
   "execution_count": 14
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
