{
 "cells": [
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "In this notebook the first section shows different libraries for reading PDF files.\n",
    "\n",
    "This was influenced by [Blueprints for Text Analytics Using Python](https://www.oreilly.com/library/view/blueprints-for-text/9781492074076/) chapters 3 `Extracting Data`\n",
    "\n",
    "The remaining sections are influenced by chapter 4 `Preparing Textual Data for Statistics and Machine Learning`\n",
    "\n",
    "# Extracting Data from PDF files"
   ],
   "id": "7a0852b5f09ce83a"
  },
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2026-02-10T06:41:09.055465Z",
     "start_time": "2026-02-10T06:41:02.099104Z"
    }
   },
   "source": [
    "\n",
    "import fhirclient.models.binary as bin\n",
    "import json\n",
    "\n",
    "from pypdf import PdfReader\n",
    "import pymupdf\n",
    "import tabula\n",
    "import base64\n",
    "import io\n",
    "import pymupdf4llm\n",
    "import pathlib\n",
    "\n",
    "report_list = ['ORU_R01_DLIMS.txt','ORU_R01_R125.1_R0A.txt','ORU_R01_R125.1_RBS.txt','ORU_R01_R125.1_REP.txt','ORU_R01_R125.1_RR8.txt','ORU_R01_R125.1_RX1.txt','ORU_R01_R125.1_SG9.txt','ORU_R01_R125.1_ZT001.txt','ORU_R01_R125.1_7A3.txt','ORU_R01_R125.1_RPY.txt','ORU_R01_R125.1_RXK.txt', 'SHIRE_ORU_R01_RM3.txt', 'WALES_ORU_R01_TX.txt']\n",
    "\n",
    "# For testing\n",
    "\n",
    "report_list = ['ORU_R01_R125.1_R0A.txt','ORU_R01_DLIMS.txt']\n",
    "\n",
    "\n",
    "pdfReaderEnabled = False\n",
    "pdfPymupdfEnabled = False\n",
    "pdfpymupdf4llmEnabled = True\n",
    "pdfTableEnabled = False\n",
    "\n",
    "\n",
    "for file in report_list:\n",
    "    with open('Output/FHIR/R01/' + file+ '.json', 'rb') as g:\n",
    "        rawJson = g.read()\n",
    "        jsonBundle = json.loads(rawJson)\n",
    "\n",
    "        #bundle = b.Bundle(jsonBundle)\n",
    "        for entry in jsonBundle['entry']:\n",
    "            #print(entry['resource']['resourceType'])\n",
    "            if entry['resource']['resourceType'] == 'Binary':\n",
    "                binary = bin.Binary(entry['resource'])\n",
    "                encoded = binary.data\n",
    "                #print(encoded)\n",
    "                decode = base64.b64decode(encoded)\n",
    "                pdf_stream = io.BytesIO(decode)\n",
    "                if pdfpymupdf4llmEnabled:\n",
    "                    doc = pymupdf.open(\"pdf\", pdf_stream)\n",
    "                    md_text = pymupdf4llm.to_markdown(doc)\n",
    "                    #print(md_text)\n",
    "                    pathlib.Path('Output/Markdown/R01/' + file+ '.md').write_bytes(md_text.encode())\n",
    "                if pdfPymupdfEnabled:\n",
    "                    # https://pymupdf.readthedocs.io/en/latest/the-basics.html\n",
    "\n",
    "                    # Initial Tests indicate this is the best method to use.\n",
    "\n",
    "                    doc = pymupdf.open(\"pdf\", pdf_stream)\n",
    "                    print('Pages: '+ str(len(doc)))\n",
    "                    print('============ pymupdf ' + file + ' ==============')\n",
    "                    for page in doc: # iterate the document pages\n",
    "                        text = page.get_text().encode(\"utf8\")\n",
    "                        print(text)\n",
    "                if pdfReaderEnabled:\n",
    "                    pdf = PdfReader(pdf_stream)\n",
    "                    print('Pages: '+ str(len(pdf.pages)))\n",
    "                    print('============ PDF Reader ' + file + ' ==============')\n",
    "                    for page in pdf.pages:\n",
    "                        text = page.extract_text()\n",
    "                        # print(text)\n",
    "                if pdfTableEnabled:\n",
    "                    tables = tabula.read_pdf(pdf_stream\n",
    "                                             , pages='all', multiple_tables=True)\n",
    "                    print('============ Tabula ' + file + ' ==============')\n",
    "                    for table in tables:\n",
    "                        tab1 = table\n",
    "                        #print(table)\n",
    "                with open('Output/PDF/R01/' + file+ '.pdf', 'wb') as pdf:\n",
    "                    pdf.write(decode)\n",
    "\n",
    "\n",
    "pdf_list = ['R24-0WH7-1.pdf','R24-1A4W-1.pdf']\n",
    "\n",
    "for file in pdf_list:\n",
    "    with open('Output/PDF/R01/' + file, 'rb') as g:\n",
    "        pdf_stream = g.read()\n",
    "        if pdfpymupdf4llmEnabled:\n",
    "            doc = pymupdf.open(\"pdf\", pdf_stream)\n",
    "            md_text = pymupdf4llm.to_markdown(doc)\n",
    "            print(md_text)\n",
    "            pathlib.Path('Output/Markdown/R01/' + file+ '.md').write_bytes(md_text.encode())\n",
    "\n",
    "\n",
    "pdf_list = ['genomic-medicine-device-test-order-form-cancer-v1.22.pdf']\n",
    "\n",
    "for file in pdf_list:\n",
    "    with open('Output/PDF/O21/' + file, 'rb') as g:\n",
    "        pdf_stream = g.read()\n",
    "        if pdfpymupdf4llmEnabled:\n",
    "            doc = pymupdf.open(\"pdf\", pdf_stream)\n",
    "            md_text = pymupdf4llm.to_markdown(doc)\n",
    "            print(md_text)\n",
    "            pathlib.Path('Output/Markdown/O21/' + file+ '.md').write_bytes(md_text.encode())\n",
    "\n"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NHS Genomic Medicine Service, WGS Test Request Cancer, July 2024 v1.22 to be used for WGS go-live.\n",
      "This document is subject to version control and is regularly updated. Please confirm you are using the current version by\n",
      "contacting your local Genomic Laboratory Hub\n",
      "\n",
      "~~**Genomic Medicine Service**~~\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "|Whole Genome Sequencing (WGS) Test Request PLEASE DO NOT USE FOR NON-WGS TESTS CANCER|Col2|\n",
      "|---|---|\n",
      "|**Requesting organisation:**<br> <br>|**Requesting organisation:**<br> <br>|\n",
      "|<br>~~**GLH laboratory to receive sample:**~~<br>North West|~~Test Required~~<br>**Whole Genome Sequencing **|\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "|GLH laboratory to receive sample: North West|Col2|Col3|Col4|Test Required Whole Genome Sequencing|Col6|Col7|\n",
      "|---|---|---|---|---|---|---|\n",
      "|~~Patient first name~~<br> <br>Myrcella|~~Patient first name~~<br> <br>Myrcella|~~Patient first name~~<br> <br>Myrcella|~~Patient first name~~<br> <br>Myrcella|~~Ethnicity~~<br>|~~Ethnicity~~<br>|~~Ethnicity~~<br>|\n",
      "|~~Patient last name~~<br> <br> <br>Wrexham|~~Patient last name~~<br> <br> <br>Wrexham|~~Patient last name~~<br> <br> <br>Wrexham|~~Patient last name~~<br> <br> <br>Wrexham|~~Test Directory Clinical Indication & code (cancer type &~~<br>~~sub-type)~~<br>**_The clinical indications listed at the bottom of the pick_**<br>**_list under 'NEW INDICATIONS' are not live for all NHS_**<br>**_GLHs. Please check with GLHs prior to ordering._**|~~Test Directory Clinical Indication & code (cancer type &~~<br>~~sub-type)~~<br>**_The clinical indications listed at the bottom of the pick_**<br>**_list under 'NEW INDICATIONS' are not live for all NHS_**<br>**_GLHs. Please check with GLHs prior to ordering._**|~~Test Directory Clinical Indication & code (cancer type &~~<br>~~sub-type)~~<br>**_The clinical indications listed at the bottom of the pick_**<br>**_list under 'NEW INDICATIONS' are not live for all NHS_**<br>**_GLHs. Please check with GLHs prior to ordering._**|\n",
      "|~~Date of birth~~_(dd/mm/yyyy)_ <br><br>16/10/1970|~~Date of birth~~_(dd/mm/yyyy)_ <br><br>16/10/1970|~~Hospital number~~<br>12345678|~~Hospital number~~<br>12345678|~~Hospital number~~<br>12345678|~~Hospital number~~<br>12345678|~~Hospital number~~<br>12345678|\n",
      "|~~Gender~~<br> Male<br> Female<br> Other<br><br>|~~Gender~~<br> Male<br> Female<br> Other<br><br>|~~Gender~~<br> Male<br> Female<br> Other<br><br>|~~Gender~~<br> Male<br> Female<br> Other<br><br>|Presentation status<br> First diagnosis<br> Recurrence / Relapse<br> Unknown <br>|Presentation status<br> First diagnosis<br> Recurrence / Relapse<br> Unknown <br>|Presentation status<br> First diagnosis<br> Recurrence / Relapse<br> Unknown <br>|\n",
      "|Postcode<br> <br>N<br>G<br>1<br>0<br>1<br>n<br>p|Postcode<br> <br>N<br>G<br>1<br>0<br>1<br>n<br>p|Postcode<br> <br>N<br>G<br>1<br>0<br>1<br>n<br>p|Postcode<br> <br>N<br>G<br>1<br>0<br>1<br>n<br>p|~~Additional clinical information (if required)~~<br>_E.g. previous tumours, molecular testing, and relevant treatment history with_<br>_date(s)_<br>Some clinical information|~~Additional clinical information (if required)~~<br>_E.g. previous tumours, molecular testing, and relevant treatment history with_<br>_date(s)_<br>Some clinical information|~~Additional clinical information (if required)~~<br>_E.g. previous tumours, molecular testing, and relevant treatment history with_<br>_date(s)_<br>Some clinical information|\n",
      "|~~NHS number~~<br>~~Reason NHS Number not availab~~le:<br> Patient not eligible for NHS number (e.g. foreign national)<br> Other (provide reason):<br>|~~NHS number~~<br>~~Reason NHS Number not availab~~le:<br> Patient not eligible for NHS number (e.g. foreign national)<br> Other (provide reason):<br>|~~NHS number~~<br>~~Reason NHS Number not availab~~le:<br> Patient not eligible for NHS number (e.g. foreign national)<br> Other (provide reason):<br>|~~NHS number~~<br>~~Reason NHS Number not availab~~le:<br> Patient not eligible for NHS number (e.g. foreign national)<br> Other (provide reason):<br>|~~NHS number~~<br>~~Reason NHS Number not availab~~le:<br> Patient not eligible for NHS number (e.g. foreign national)<br> Other (provide reason):<br>|~~NHS number~~<br>~~Reason NHS Number not availab~~le:<br> Patient not eligible for NHS number (e.g. foreign national)<br> Other (provide reason):<br>|~~NHS number~~<br>~~Reason NHS Number not availab~~le:<br> Patient not eligible for NHS number (e.g. foreign national)<br> Other (provide reason):<br>|\n",
      "|~~**Solid tumour requests only**~~<br> <br> <br>|~~**Solid tumour requests only**~~<br> <br> <br>|~~**Solid tumour requests only**~~<br> <br> <br>|~~**Solid tumour requests only**~~<br> <br> <br>|~~**Solid tumour requests only**~~<br> <br> <br>|~~**Solid tumour requests only**~~<br> <br> <br>|~~**Solid tumour requests only**~~<br> <br> <br>|\n",
      "|~~Primary~~<br> Metastatic<br> Unknown<br> Lymphoma<br> <br>\u0014|~~Histopathology Lab ID~~<br> <br>1234567PATH|~~Histopathology Lab ID~~<br> <br>1234567PATH|~~Histopathology Lab ID~~<br> <br>1234567PATH|~~Additional tumour information (if relevant)~~<br>_E.g. site of metastasis (if metastatic), or unknown primary_<br> <br>|~~Additional tumour information (if relevant)~~<br>_E.g. site of metastasis (if metastatic), or unknown primary_<br> <br>|~~Additional tumour information (if relevant)~~<br>_E.g. site of metastasis (if metastatic), or unknown primary_<br> <br>|\n",
      "|~~Primary~~<br> Metastatic<br> Unknown<br> Lymphoma<br> <br>\u0014|~~Date of this diagnosis~~_(dd/mm/yyyy)_ <br>|~~Date of this diagnosis~~_(dd/mm/yyyy)_ <br>|~~Date of this diagnosis~~_(dd/mm/yyyy)_ <br>|~~Tumour topography ~~|~~Tumour morphology ~~|~~Tumour morphology ~~|\n",
      "|~~**Haemato-oncology liquid tumour requests only**~~<br> <br> <br> <br> <br>|~~**Haemato-oncology liquid tumour requests only**~~<br> <br> <br> <br> <br>|~~**Haemato-oncology liquid tumour requests only**~~<br> <br> <br> <br> <br>|~~**Haemato-oncology liquid tumour requests only**~~<br> <br> <br> <br> <br>|~~**Haemato-oncology liquid tumour requests only**~~<br> <br> <br> <br> <br>|~~**Haemato-oncology liquid tumour requests only**~~<br> <br> <br> <br> <br>|~~**Haemato-oncology liquid tumour requests only**~~<br> <br> <br> <br> <br>|\n",
      "|~~ AML~~<br>~~ ALL~~<br>~~ Other (please specify):~~<br>|~~ AML~~<br>~~ ALL~~<br>~~ Other (please specify):~~<br>|~~ AML~~<br>~~ ALL~~<br>~~ Other (please specify):~~<br>|~~ AML~~<br>~~ ALL~~<br>~~ Other (please specify):~~<br>|~~ SIHMDS Lab ID~~<br>|~~Date of this diagnosis~~_(dd/mm/yyyy)_|~~Date of this diagnosis~~_(dd/mm/yyyy)_|\n",
      "|~~**Complete for tumour samples**~~~~_(being sent to GLH DNA extraction lab)_~~ <br> <br> <br> <br>|~~**Complete for tumour samples**~~~~_(being sent to GLH DNA extraction lab)_~~ <br> <br> <br> <br>|~~**Complete for tumour samples**~~~~_(being sent to GLH DNA extraction lab)_~~ <br> <br> <br> <br>|~~**Complete for tumour samples**~~~~_(being sent to GLH DNA extraction lab)_~~ <br> <br> <br> <br>|~~**Complete for tumour samples**~~~~_(being sent to GLH DNA extraction lab)_~~ <br> <br> <br> <br>|~~**Complete for tumour samples**~~~~_(being sent to GLH DNA extraction lab)_~~ <br> <br> <br> <br>|~~**Complete for tumour samples**~~~~_(being sent to GLH DNA extraction lab)_~~ <br> <br> <br> <br>|\n",
      "|~~ Fresh frozen tumour~~<br>~~ Bone marrow~~<br>~~ Blood (EDTA)~~<br>~~ Other (please specify):~~<br> <br>\u0014|~~ Fresh frozen tumour~~<br>~~ Bone marrow~~<br>~~ Blood (EDTA)~~<br>~~ Other (please specify):~~<br> <br>\u0014|~~ Fresh frozen tumour~~<br>~~ Bone marrow~~<br>~~ Blood (EDTA)~~<br>~~ Other (please specify):~~<br> <br>\u0014|~~ Fresh frozen tumour~~<br>~~ Bone marrow~~<br>~~ Blood (EDTA)~~<br>~~ Other (please specify):~~<br> <br>\u0014|~~ Fresh frozen tumour~~<br>~~ Bone marrow~~<br>~~ Blood (EDTA)~~<br>~~ Other (please specify):~~<br> <br>\u0014|~~ Fresh frozen tumour~~<br>~~ Bone marrow~~<br>~~ Blood (EDTA)~~<br>~~ Other (please specify):~~<br> <br>\u0014|~~ Fresh frozen tumour~~<br>~~ Bone marrow~~<br>~~ Blood (EDTA)~~<br>~~ Other (please specify):~~<br> <br>\u0014|\n",
      "|<br> <br> <br> <br>~~_% malignant nuclei / blasts or equivalent in this sample (refer to sample handling guidance) must be provided below_~~<br> <br> <br> <br>|<br> <br> <br> <br>~~_% malignant nuclei / blasts or equivalent in this sample (refer to sample handling guidance) must be provided below_~~<br> <br> <br> <br>|<br> <br> <br> <br>~~_% malignant nuclei / blasts or equivalent in this sample (refer to sample handling guidance) must be provided below_~~<br> <br> <br> <br>|<br> <br> <br> <br>~~_% malignant nuclei / blasts or equivalent in this sample (refer to sample handling guidance) must be provided below_~~<br> <br> <br> <br>|<br> <br> <br> <br>~~_% malignant nuclei / blasts or equivalent in this sample (refer to sample handling guidance) must be provided below_~~<br> <br> <br> <br>|<br> <br> <br> <br>~~_% malignant nuclei / blasts or equivalent in this sample (refer to sample handling guidance) must be provided below_~~<br> <br> <br> <br>|<br> <br> <br> <br>~~_% malignant nuclei / blasts or equivalent in this sample (refer to sample handling guidance) must be provided below_~~<br> <br> <br> <br>|\n",
      "|~~ Sample ID~~|~~ Sample ID~~|~~ Sample ID~~|~~Collection date / time~~|~~% Malignant nuclei / blasts~~|~~% Malignant nuclei / blasts~~|~~If BM/PB provide volume~~<br>and nucleated cell count|\n",
      "|<br>SAMPLEID1|<br>SAMPLEID1|<br>SAMPLEID1|||||\n",
      "|~~**Complete for germline samples**~~~~_(being sent to GLH DNA extraction lab)_ ~~<br> <br> <br> <br> <br>|~~**Complete for germline samples**~~~~_(being sent to GLH DNA extraction lab)_ ~~<br> <br> <br> <br> <br>|~~**Complete for germline samples**~~~~_(being sent to GLH DNA extraction lab)_ ~~<br> <br> <br> <br> <br>|~~**Complete for germline samples**~~~~_(being sent to GLH DNA extraction lab)_ ~~<br> <br> <br> <br> <br>|~~**Complete for germline samples**~~~~_(being sent to GLH DNA extraction lab)_ ~~<br> <br> <br> <br> <br>|~~**Complete for germline samples**~~~~_(being sent to GLH DNA extraction lab)_ ~~<br> <br> <br> <br> <br>|~~**Complete for germline samples**~~~~_(being sent to GLH DNA extraction lab)_ ~~<br> <br> <br> <br> <br>|\n",
      "|~~ Blood (EDTA)~~<br>~~ Saliva~~<br>~~ Fibroblasts~~<br>~~ Skin biopsy~~<br>~~ Other (please specify):~~<br> <br> <br> <br>|~~ Blood (EDTA)~~<br>~~ Saliva~~<br>~~ Fibroblasts~~<br>~~ Skin biopsy~~<br>~~ Other (please specify):~~<br> <br> <br> <br>|~~ Blood (EDTA)~~<br>~~ Saliva~~<br>~~ Fibroblasts~~<br>~~ Skin biopsy~~<br>~~ Other (please specify):~~<br> <br> <br> <br>|~~ Blood (EDTA)~~<br>~~ Saliva~~<br>~~ Fibroblasts~~<br>~~ Skin biopsy~~<br>~~ Other (please specify):~~<br> <br> <br> <br>|~~ Blood (EDTA)~~<br>~~ Saliva~~<br>~~ Fibroblasts~~<br>~~ Skin biopsy~~<br>~~ Other (please specify):~~<br> <br> <br> <br>|~~ Blood (EDTA)~~<br>~~ Saliva~~<br>~~ Fibroblasts~~<br>~~ Skin biopsy~~<br>~~ Other (please specify):~~<br> <br> <br> <br>|~~ Blood (EDTA)~~<br>~~ Saliva~~<br>~~ Fibroblasts~~<br>~~ Skin biopsy~~<br>~~ Other (please specify):~~<br> <br> <br> <br>|\n",
      "|~~Sample ID~~|~~Sample ID~~|~~Sample ID~~|~~Collection date / time~~|~~Sample volume if applicable~~|~~Sample volume if applicable~~|~~Comments~~|\n",
      "||||||||\n",
      "|~~**Responsible consultant **~~|~~**Responsible consultant **~~|~~**Responsible consultant **~~|~~**Responsible consultant **~~|<br>~~**Main contact (if different from responsible consultant)**~~|<br>~~**Main contact (if different from responsible consultant)**~~|<br>~~**Main contact (if different from responsible consultant)**~~|\n",
      "|<br>Name:<br>Department address:<br>Phone:<br>Email:|<br>Name:<br>Department address:<br>Phone:<br>Email:|<br>Name:<br>Department address:<br>Phone:<br>Email:|<br>Name:<br>Department address:<br>Phone:<br>Email:|~~Name:~~<br>Department address:<br>Phone:<br>Email:<br>|~~Name:~~<br>Department address:<br>Phone:<br>Email:<br>|~~Name:~~<br>Department address:<br>Phone:<br>Email:<br>|\n",
      "\n",
      "\n",
      "Patient conversation taken place; Record of Discussion form to follow\n",
      "\n",
      "\n",
      "Version 1.22 Page 1 of 1\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "execution_count": 17
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "## Preparing Textual Data for Statistics and Machine Learning\n",
    "\n",
    "### Extracting Data from a Laboratory Report"
   ],
   "id": "3d4bfbf14579d2a"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-02-10T06:41:09.841803Z",
     "start_time": "2026-02-10T06:41:09.057423Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import markdown\n",
    "\n",
    "# Karotype testing\n",
    "file = 'R24-0WH7-1.pdf'\n",
    "\n",
    "# this is typical of reports coming at present from iGene\n",
    "file = 'ORU_R01_R125.1_R0A.txt.pdf'\n",
    "\n",
    "folder = 'Output/PDF/R01/'\n",
    "\n",
    "with open(folder + file, 'rb') as g:\n",
    "    pdf_stream = g.read()\n",
    "\n",
    "    doc = pymupdf.open(\"pdf\", pdf_stream)\n",
    "    md_text = pymupdf4llm.to_markdown(doc)\n",
    "    html_text = markdown.markdown(md_text)\n"
   ],
   "id": "abf2dd742cdc72ae",
   "outputs": [],
   "execution_count": 18
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### Cleaning Text Data",
   "id": "d146e23724f48c4d"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-02-10T07:11:18.880955Z",
     "start_time": "2026-02-10T07:11:18.854557Z"
    }
   },
   "cell_type": "code",
   "source": [
    "from io import StringIO\n",
    "from html.parser import HTMLParser\n",
    "import re\n",
    "\n",
    "class MLStripper(HTMLParser):\n",
    "    def __init__(self):\n",
    "        super().__init__()\n",
    "        self.reset()\n",
    "        self.strict = False\n",
    "        self.convert_charrefs= True\n",
    "        self.text = StringIO()\n",
    "    def handle_data(self, d):\n",
    "        self.text.write(d)\n",
    "    def get_data(self):\n",
    "        return self.text.getvalue()\n",
    "\n",
    "def strip_tags(html):\n",
    "    s = MLStripper()\n",
    "    s.feed(html)\n",
    "    return s.get_data()\n",
    "\n",
    "def getItem(field_name, text):\n",
    "    pattern = rf\"{re.escape(field_name)}:*([^\\r\\n]+)\"\n",
    "    match = re.search(pattern, text)\n",
    "    if match:\n",
    "        report_no = match.group(1)\n",
    "        return report_no.strip()\n",
    "    return \"\"\n",
    "\n",
    "def getItemWithStopWord(field_name, stop_word, text):\n",
    "\n",
    "    pattern = rf\"{re.escape(field_name)}:*([^\\r\\n]+?)({re.escape(stop_word)})\"\n",
    "    match = re.search(pattern, text)\n",
    "    if match:\n",
    "        report_no = match.group(1)\n",
    "        return report_no.strip()\n",
    "\n",
    "    return getItem(field_name, text)\n",
    "\n",
    "clean_text = strip_tags(html_text)\n",
    "\n",
    "print(\"Report No = \"+getItem('Report No', clean_text))\n",
    "print(\"Hospital No = \"+getItem('Hospital No', clean_text))\n",
    "print(\"NHS No = \"+getItemWithStopWord('NHS No','Hive', clean_text))\n",
    "print(\"Order Number (HIVE) = \"+getItemWithStopWord('Hive Order ID','Path', clean_text))\n",
    "print(\"Sample type = \"+getItemWithStopWord('Sample','Sample', clean_text))\n",
    "print(\"Sample2 type = \"+getItem('Sample2', clean_text))\n",
    "print(\"Pathology No. = \"+getItem('Path. No', clean_text))\n",
    "print(\"Specimen Collected = \"+getItemWithStopWord('Collected','Activated', clean_text))\n",
    "print(\"Specimen Collected2 = \"+getItemWithStopWord('Collected2','Received2', clean_text))\n",
    "print(\"Specimen Received2 = \"+getItemWithStopWord('Received2','Received2', clean_text))"
   ],
   "id": "2a8570843bdb03ef",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Report No = R25-00EK-3\n",
      "Hospital No = Not provided\n",
      "NHS No = :\n",
      "Order Number (HIVE) = 1000152866\n",
      "Sample type = Blood\n",
      "Sample2 type = \n",
      "Pathology No. = ACC1810A1\n",
      "Specimen Collected = Not provided\n",
      "Specimen Collected2 = \n",
      "Specimen Received2 = \n"
     ]
    }
   ],
   "execution_count": 41
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### [spaCy](https://spacy.io/)",
   "id": "dcee0858a73992e9"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-02-10T06:41:10.871156Z",
     "start_time": "2026-02-10T06:41:09.903543Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import spacy\n",
    "\n",
    "\n",
    "print('============ Spacy ' + file + ' ==============')\n",
    "nlp = spacy.load('en_core_web_sm')\n",
    "\n",
    "#doc = pdf_reader(folder + file, nlp)\n",
    "doc = nlp(clean_text)\n",
    "\n",
    "for token in doc:\n",
    "    print(token,end=\"|\")"
   ],
   "id": "c82b5bb90642d50d",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "North|West|Genomic|Laboratory|Hub|(|Manchester|Site|)|\n",
      "|Manchester|Centre|for|Genomic|Medicine|\n",
      "|6th|Floor|,|St|Mary|'s|Hospital|,|Manchester|,|M13|9WL|\n",
      "|Scientific|Operational|Director|:|Dr|E.|Howard|\n",
      "|https://mft.nhs.uk/nwglh/|mft.genomics@nhs.net|Tel|+44(0|)|161|276|6122|\n",
      "|Forename|:|Jack|\n",
      "|Surname|:|HIM|\n",
      "|Sex|:|\n",
      "|Report|No|:|R25|-|00EK-3|\n",
      "|Hospital|No|:|Not|provided|\n",
      "|Other|No|:|Not|provided|\n",
      "|DoB|:|18/02/1987|Sample|:|Blood|\n",
      "|NHS|No|:|\n",
      "|Hive|Order|ID|:|1000152866|Path|.|No|:|ACC1810A1|\n",
      "|Collected|:|Not|provided|Activated|:|14/10/2025|\n",
      "|Reported|:|14/10/2025|\n",
      "|Referred|by|:|Jonathan|Edgerley|,|Clinical|Genetics|,|North|West|Genomic|Laboratory|Hub|(|Manchester|)|,|Manchester|,|M13|9WL|,|(|jonathan|.|\n",
      "|edgerley@mft.nhs.uk|)|\n",
      "|Genomics|Laboratory|Report|\n",
      "|Reason|for|Testing|\n",
      "|[|REASON|FOR|TESTING|-|COPY|BELOW|TEXT|TO|'|REPORTING|REFERRAL|INFORMATION|TEXT|'|BOX|THEN|DELETE|]|\n",
      "|[|Insert|referring|clinician|notes|]|\n",
      "|Clinical|Indication|:|[|Insert|Clinical|Indication|Name|(|RXXX|)|]|.|\n",
      "|RESULT|SUMMARY|:|\n",
      "|Genetic|cause|not|identified|.|\n",
      "|Result|and|Interpretation|\n",
      "|No|pathogenic|variants|that|explain|the|genetic|cause|of|this|individual|'s|clinical|symptoms|were|identified|.|\n",
      "|This|result|does|not|rule|out|the|diagnosis|of|a|genetic|disorder|since|some|DNA|abnormalities|are|undetectable|by|the|applied|technology|\n",
      "|(|see|Appendix|I|overleaf|)|.|\n",
      "|Reported|By|:|\n",
      "|Authorised|By|:|\n",
      "|Jonathan|Edgerley|\n",
      "|Clinical|Scientist|\n",
      "|Jonathan|Edgerley|\n",
      "|Clinical|Scientist|\n",
      "|Page|1|of|2|\n",
      "|Forename|:|Jack|Report|No|:|R25|-|00EK-3|\n",
      "|Surname|:|HIM|DoB|:|18/02/1987|\n",
      "|APPENDIX|I|-|Test|Methodology|:|\n",
      "|Clinical|indication|:|Thoracic|aortic|aneurysm|and|dissection|(|R125.1|,|R125.2|)|\n",
      "|Laboratory|process|\n",
      "|DNA|extraction|:|Chemagic|Prime|for|samples|extracted|by|NWGLH|\n",
      "|Enrichment|method|:|Agilent|SureSelect|custom|design|\n",
      "|Sequencing|method|:|Illumina|NextSeq|550|\n",
      "|Bioinformatics|and|quality|control|:|Congenica|–|Alignment|and|SNV|/|Indel|calling|–|Sentieon|DNASeq|;|CNV|calling|–|Exome|Depth|\n",
      "|Panel(s|)|applied|:|Thoracic|aortic|aneurysm|and|dissection|v1.2|(|R125.1|,|R125.2|)|\n",
      "|ABL1|,|ACTA2|,|BGN|,|CBS|,|COL3A1|,|COL5A1|,|COL5A2|,|EFEMP2|,|ELN|,|FBLN5|,|FBN1|,|FBN2|,|FKBP14|,|FLNA|,|FOXE3|,|LOX|,|MFAP5|,|MYH11|,|MYLK|,|NOTCH1|,|PLOD1|,|\n",
      "|PRKG1|,|SKI|,|SLC2A10|,|SMAD2|,|SMAD3|,|SMAD4|,|SMAD6|,|TGFB2|,|TGFB3|,|TGFBR1|,|TGFBR2|.|\n",
      "|The|panel|includes|diagnostic|grade|(|green|)|genes|from|the|associated|PanelApp|panel|(|https://nhsgms-panelapp.genomicsengland.co.uk/|)|for|the|specific|version|detailed|\n",
      "|above|.|This|test|is|used|to|detect|single|nucleotide|variants|(|SNVs|)|,|small|insertions|and|deletions|(|INDELs|)|and|copy|number|variations|(|defined|as|single|exon|or|larger|\n",
      "|deletions|and|duplications|)|.|This|panel|targets|protein|coding|exons|,|exon|intron|boundaries|(|±|8|bps|)|and|selected|non|-|coding|,|deep|intronic|variants|.|Only|variants|relevant|\n",
      "|to|this|patient|’s|submitted|phenotype|have|been|reported|.|If|the|result|does|not|fully|account|for|the|phenotype|observed|in|the|patient|,|please|contact|the|laboratory|to|\n",
      "|discuss|if|further|analysis|may|be|appropriate|.|\n",
      "|Coverage|:|>|97|%|of|the|coding|regions|of|the|diagnostic|grade|(|green|)|genes|in|the|panel|above|are|covered|to|a|minimum|depth|of|50X.|Patient|specific|coverage|for|\n",
      "|individual|genes|can|be|provided|on|request|.|\n",
      "|Limitations|:|This|test|can|not|be|used|for|the|detection|of|repeat|expansions|or|variants|in|mitochondrial|DNA|(|mtDNA|)|.|The|test|does|not|identify|balanced|translocations|or|\n",
      "|complex|inversions|,|and|it|may|not|detect|low|-|level|mosaicism|or|inter|-|locus|gene|conversion|.|The|sensitivity|to|detect|variants|may|be|limited|in|genes|where|some|,|or|all|,|of|\n",
      "|the|gene|is|duplicated|in|the|genome|or|the|gene|has|suboptimal|coverage|.|Specific|regions|for|R125|include|CBS|exons|3|-|17|(|NM_000071.2|)|.|Please|contact|the|laboratory|\n",
      "|for|additional|details|.|\n",
      "|Confirmation|of|findings|:|SNV|and|INDEL|calls|which|do|not|meet|internally|validated|quality|metrics|are|confirmed|by|Sanger|sequencing|.|The|presence|,|but|not|the|\n",
      "|extent|,|of|CNV|calls|is|confirmed|by|an|independent|method|(|MLPA|,|custom|droplet|digital|PCR|assay|,|or|custom|QF|-|PCR|assay|)|.|\n",
      "|Analytical|validation|:|This|laboratory|-|developed|test|has|been|locally|validated|.|The|sensitivity|of|this|methodology|for|SNVs|is|100|%|(|95|%|CI|:|99.78|%|to|100|%|)|,|INDELs|is|\n",
      "|95.86|%|(|95|%|CI|:|91.65|%|to|98.32|%|)|and|CNVs|is|100|%|(|95|%|CI|:|93.28|%|to|100|%|)|,|and|specificity|is|>|99.9|%|for|most|variant|types|.|A|normal|result|does|not|rule|out|the|\n",
      "|diagnosis|of|a|genetic|disorder|since|some|DNA|abnormalities|are|undetectable|by|the|applied|technology|.|\n",
      "|Regulation|and|accreditations|:|This|test|is|not|UKAS|accredited|.|This|test|has|been|validated|locally|and|will|be|submitted|for|UKAS|assessment|imminently|.|\n",
      "|Please|contact|the|laboratory|for|additional|details|,|if|required|.|\n",
      "|Page|2|of|2|"
     ]
    }
   ],
   "execution_count": 20
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### Extracting Named Entities",
   "id": "a14597d207d473fe"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-02-10T06:41:10.934560Z",
     "start_time": "2026-02-10T06:41:10.897155Z"
    }
   },
   "cell_type": "code",
   "source": [
    "\n",
    " for ent in doc.ents:\n",
    "    print(ent.text, ent.label_)"
   ],
   "id": "88640e4c07362bb1",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "North West Genomic Laboratory Hub ORG\n",
      "Manchester Site PERSON\n",
      "Manchester Centre PERSON\n",
      "6th Floor ORG\n",
      "St Mary's PERSON\n",
      "Manchester GPE\n",
      "M13 PERSON\n",
      "Scientific Operational ORG\n",
      "E. Howard PERSON\n",
      "Tel PERSON\n",
      "161 CARDINAL\n",
      "276 6122 DATE\n",
      "Jack\n",
      "Surname PERSON\n",
      "Hive Order PERSON\n",
      "1000152866 CARDINAL\n",
      "Jonathan Edgerley PERSON\n",
      "Clinical Genetics PERSON\n",
      "North West Genomic Laboratory Hub ORG\n",
      "Manchester PERSON\n",
      "Manchester PERSON\n",
      "M13 PERSON\n",
      "jonathan PERSON\n",
      "Genomics Laboratory Report\n",
      "Reason for Testing ORG\n",
      "Insert PERSON\n",
      "Insert Clinical Indication Name ORG\n",
      "Appendix PERSON\n",
      "Jonathan Edgerley\n",
      "Clinical Scientist PERSON\n",
      "Jonathan Edgerley\n",
      "Clinical Scientist\n",
      "Page PERSON\n",
      "1 CARDINAL\n",
      "2 CARDINAL\n",
      "18/02/1987 DATE\n",
      "Thoracic ORG\n",
      "R125.1 NORP\n",
      "R125.2 GPE\n",
      "Chemagic ORG\n",
      "NWGLH\n",
      "Enrichment ORG\n",
      "550 CARDINAL\n",
      "Congenica NORP\n",
      "Sentieon PERSON\n",
      "CNV ORG\n",
      "Exome Depth PERSON\n",
      "Thoracic ORG\n",
      "R125.1 NORP\n",
      "R125.2 GPE\n",
      "ABL1 PERSON\n",
      "ACTA2 PERSON\n",
      "BGN ORG\n",
      "CBS ORG\n",
      "EFEMP2 PRODUCT\n",
      "FBN1 ORG\n",
      "FBN2 ORG\n",
      "FLNA ORG\n",
      "FOXE3 ORG\n",
      "LOX ORG\n",
      "MFAP5 GPE\n",
      "MYLK ORG\n",
      "NOTCH1 ORG\n",
      "PLOD1 ORG\n",
      "PRKG1 ORG\n",
      "SKI ORG\n",
      "SMAD2, SMAD3, ORG\n",
      "SMAD4, SMAD6 ORG\n",
      "TGFB2 ORG\n",
      "TGFB3 GPE\n",
      "TGFBR1 PRODUCT\n",
      "TGFBR2 PERSON\n",
      "PanelApp ORG\n",
      "INDELs ORG\n",
      "8 CARDINAL\n",
      "97% PERCENT\n",
      "R125 PRODUCT\n",
      "CBS ORG\n",
      "3 CARDINAL\n",
      "INDEL ORG\n",
      "Sanger PERSON\n",
      "CNV ORG\n",
      "100% PERCENT\n",
      "95% PERCENT\n",
      "99.78% to 100% PERCENT\n",
      "INDELs ORG\n",
      "95.86% PERCENT\n",
      "95% PERCENT\n",
      "CI ORG\n",
      "91.65% PERCENT\n",
      "98.32% PERCENT\n",
      "CNVs ORG\n",
      "100% PERCENT\n",
      "95% PERCENT\n",
      "93.28% to 100% PERCENT\n",
      "99.9% PERCENT\n",
      "UKAS NORP\n",
      "UKAS NORP\n",
      "2 CARDINAL\n",
      "2 CARDINAL\n"
     ]
    }
   ],
   "execution_count": 21
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-02-10T06:41:10.978185Z",
     "start_time": "2026-02-10T06:41:10.936988Z"
    }
   },
   "cell_type": "code",
   "source": [
    "from spacy import displacy\n",
    "\n",
    "displacy.render(doc, style='ent', jupyter=False)"
   ],
   "id": "15689d79a00ec9bf",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'<div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">\\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    North West Genomic Laboratory Hub\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n (\\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Manchester Site\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n)<br>\\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Manchester Centre\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n for Genomic Medicine<br>\\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    6th Floor\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    St Mary\\'s\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n Hospital, \\n<mark class=\"entity\" style=\"background: #feca74; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Manchester\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">GPE</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    M13\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n 9WL<br>\\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Scientific Operational\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n Director: Dr \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    E. Howard\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n<br>https://mft.nhs.uk/nwglh/ mft.genomics@nhs.net \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Tel\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n +44(0) \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    161\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #bfe1d9; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    276 6122\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">DATE</span>\\n</mark>\\n<br>Forename: \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Jack\\nSurname\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n: HIM<br>Sex:<br>Report No: R25-00EK-3<br>Hospital No: Not provided<br>Other No: Not provided<br>DoB: 18/02/1987 Sample: Blood<br>NHS No:<br>\\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Hive Order\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n ID: \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    1000152866\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\\n</mark>\\n Path. No: ACC1810A1<br>Collected: Not provided Activated: 14/10/2025<br>Reported: 14/10/2025<br>Referred by: \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Jonathan Edgerley\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Clinical Genetics\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    North West Genomic Laboratory Hub\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n (\\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Manchester\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n), \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Manchester\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    M13\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n 9WL, (\\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    jonathan\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n.<br>edgerley@mft.nhs.uk)<br>\\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Genomics Laboratory Report\\nReason for Testing\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n<br>[REASON FOR TESTING - COPY BELOW TEXT TO \\'REPORTING REFERRAL INFORMATION TEXT\\' BOX THEN DELETE]<br>[\\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Insert\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n referring clinician notes]<br>Clinical Indication: [\\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Insert Clinical Indication Name\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n (RXXX)].<br>RESULT SUMMARY:<br>Genetic cause not identified.<br>Result and Interpretation<br>No pathogenic variants that explain the genetic cause of this individual\\'s clinical symptoms were identified.<br>This result does not rule out the diagnosis of a genetic disorder since some DNA abnormalities are undetectable by the applied technology<br>(see \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Appendix\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n I overleaf).<br>Reported By:<br>Authorised By:<br>\\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Jonathan Edgerley\\nClinical Scientist\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n<br>\\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Jonathan Edgerley\\nClinical Scientist\\nPage\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\\n</mark>\\n of \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\\n</mark>\\n<br>Forename: Jack Report No: R25-00EK-3<br>Surname: HIM DoB: \\n<mark class=\"entity\" style=\"background: #bfe1d9; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    18/02/1987\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">DATE</span>\\n</mark>\\n<br>APPENDIX I - Test Methodology:<br>Clinical indication: \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Thoracic\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n aortic aneurysm and dissection (\\n<mark class=\"entity\" style=\"background: #c887fb; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    R125.1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">NORP</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #feca74; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    R125.2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">GPE</span>\\n</mark>\\n)<br>Laboratory process<br>DNA extraction: \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Chemagic\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n Prime for samples extracted by \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    NWGLH\\nEnrichment\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n method: Agilent SureSelect custom design<br>Sequencing method: Illumina NextSeq \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    550\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\\n</mark>\\n<br>Bioinformatics and quality control: \\n<mark class=\"entity\" style=\"background: #c887fb; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Congenica\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">NORP</span>\\n</mark>\\n – Alignment and SNV/Indel calling – \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Sentieon\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n DNASeq; \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    CNV\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n calling – \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Exome Depth\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n<br>Panel(s) applied: \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Thoracic\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n aortic aneurysm and dissection v1.2 (\\n<mark class=\"entity\" style=\"background: #c887fb; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    R125.1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">NORP</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #feca74; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    R125.2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">GPE</span>\\n</mark>\\n)<br>\\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    ABL1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    ACTA2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    BGN\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    CBS\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n, COL3A1, COL5A1, COL5A2, \\n<mark class=\"entity\" style=\"background: #bfeeb7; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    EFEMP2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PRODUCT</span>\\n</mark>\\n, ELN, FBLN5, \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    FBN1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    FBN2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n, FKBP14, \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    FLNA\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    FOXE3\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    LOX\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #feca74; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    MFAP5\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">GPE</span>\\n</mark>\\n, MYH11, \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    MYLK\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    NOTCH1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    PLOD1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n,<br>\\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    PRKG1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    SKI\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n, SLC2A10, \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    SMAD2, SMAD3,\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    SMAD4, SMAD6\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    TGFB2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #feca74; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    TGFB3\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">GPE</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #bfeeb7; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    TGFBR1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PRODUCT</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    TGFBR2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n.<br>The panel includes diagnostic grade (green) genes from the associated \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    PanelApp\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n panel (https://nhsgms-panelapp.genomicsengland.co.uk/) for the specific version detailed<br>above. This test is used to detect single nucleotide variants (SNVs), small insertions and deletions (\\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    INDELs\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n) and copy number variations (defined as single exon or larger<br>deletions and duplications). This panel targets protein coding exons, exon intron boundaries (± \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    8\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\\n</mark>\\n bps) and selected non-coding, deep intronic variants. Only variants relevant<br>to this patient’s submitted phenotype have been reported. If the result does not fully account for the phenotype observed in the patient, please contact the laboratory to<br>discuss if further analysis may be appropriate.<br>Coverage: &gt;\\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    97%\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\\n</mark>\\n of the coding regions of the diagnostic grade (green) genes in the panel above are covered to a minimum depth of 50X. Patient specific coverage for<br>individual genes can be provided on request.<br>Limitations: This test cannot be used for the detection of repeat expansions or variants in mitochondrial DNA (mtDNA). The test does not identify balanced translocations or<br>complex inversions, and it may not detect low-level mosaicism or inter-locus gene conversion. The sensitivity to detect variants may be limited in genes where some, or all, of<br>the gene is duplicated in the genome or the gene has suboptimal coverage. Specific regions for \\n<mark class=\"entity\" style=\"background: #bfeeb7; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    R125\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PRODUCT</span>\\n</mark>\\n include \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    CBS\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n exons \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    3\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\\n</mark>\\n-17 (NM_000071.2). Please contact the laboratory<br>for additional details.<br>Confirmation of findings: SNV and \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    INDEL\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n calls which do not meet internally validated quality metrics are confirmed by \\n<mark class=\"entity\" style=\"background: #aa9cfc; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Sanger\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERSON</span>\\n</mark>\\n sequencing. The presence, but not the<br>extent, of \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    CNV\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n calls is confirmed by an independent method (MLPA, custom droplet digital PCR assay, or custom QF-PCR assay).<br>Analytical validation: This laboratory-developed test has been locally validated. The sensitivity of this methodology for SNVs is \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    100%\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\\n</mark>\\n (\\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    95%\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\\n</mark>\\n CI: \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    99.78% to 100%\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\\n</mark>\\n), \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    INDELs\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n is<br>\\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    95.86%\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\\n</mark>\\n (\\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    95%\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    CI\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n: \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    91.65%\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\\n</mark>\\n to \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    98.32%\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\\n</mark>\\n) and \\n<mark class=\"entity\" style=\"background: #7aecec; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    CNVs\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ORG</span>\\n</mark>\\n is \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    100%\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\\n</mark>\\n (\\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    95%\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\\n</mark>\\n CI: \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    93.28% to 100%\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\\n</mark>\\n), and specificity is &gt;\\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    99.9%\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">PERCENT</span>\\n</mark>\\n for most variant types. A normal result does not rule out the<br>diagnosis of a genetic disorder since some DNA abnormalities are undetectable by the applied technology.<br>Regulation and accreditations: This test is not \\n<mark class=\"entity\" style=\"background: #c887fb; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    UKAS\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">NORP</span>\\n</mark>\\n accredited. This test has been validated locally and will be submitted for \\n<mark class=\"entity\" style=\"background: #c887fb; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    UKAS\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">NORP</span>\\n</mark>\\n assessment imminently.<br>Please contact the laboratory for additional details, if required.<br>Page \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\\n</mark>\\n of \\n<mark class=\"entity\" style=\"background: #e4e7d2; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">CARDINAL</span>\\n</mark>\\n</div>'"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 22
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "## [scispaCy](https://github.com/allenai/scispacy)\n",
    "\n",
    "A full spaCy pipeline and models for scientific/biomedical documents. Note: For clinical documents consider using [medspaCy](https://spacy.io/universe/project/medspacy)"
   ],
   "id": "149a85a96b0ec842"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-02-10T06:41:39.260702Z",
     "start_time": "2026-02-10T06:41:10.992593Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import scispacy\n",
    "from scispacy.abbreviation import AbbreviationDetector\n",
    "\n",
    "import pyobo\n",
    "\n",
    "from scispacy.linking import EntityLinker\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "nlp = spacy.load(\"en_core_sci_sm\")\n",
    "\n",
    "nlp.add_pipe(\"abbreviation_detector\")\n",
    "\n",
    "doc = nlp(clean_text)\n",
    "\n",
    "print(\"Abbreviation\", \"\\t\", \"Definition\")\n",
    "for abrv in doc._.abbreviations:\n",
    "    print(f\"{abrv} \\t ({abrv.start}, {abrv.end}) {abrv._.long_form}\")\n"
   ],
   "id": "6027790bac2b23c",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Abbreviation \t Definition\n",
      "SNVs \t (483, 484) single nucleotide variants\n",
      "SNVs \t (797, 798) single nucleotide variants\n",
      "INDELs \t (813, 814) insertions and deletions\n",
      "INDELs \t (491, 492) insertions and deletions\n",
      "mtDNA \t (641, 642) mitochondrial DNA\n"
     ]
    }
   ],
   "execution_count": 23
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "Named Entities",
   "id": "349eb551b0dba1b6"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-02-10T06:41:39.321468Z",
     "start_time": "2026-02-10T06:41:39.289453Z"
    }
   },
   "cell_type": "code",
   "source": [
    " for ent in doc.ents:\n",
    "    print(ent.text,  ent.label_)"
   ],
   "id": "43b133ce0b7b872c",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "North West Genomic Laboratory Hub ENTITY\n",
      "Manchester Site ENTITY\n",
      "Genomic Medicine\n",
      "6th Floor ENTITY\n",
      "St Mary's Hospital ENTITY\n",
      "Manchester ENTITY\n",
      "M13 9WL ENTITY\n",
      "E. Howard\n",
      "https://mft.nhs.uk/nwglh/ mft.genomics@nhs.net Tel +44(0 ENTITY\n",
      "Forename ENTITY\n",
      "Jack\n",
      "Surname ENTITY\n",
      "HIM ENTITY\n",
      "Sex ENTITY\n",
      "DoB ENTITY\n",
      "Sample ENTITY\n",
      "Blood\n",
      "NHS No ENTITY\n",
      "ID ENTITY\n",
      "ACC1810A1\n",
      "Collected ENTITY\n",
      "Activated ENTITY\n",
      "Reported ENTITY\n",
      "Referred ENTITY\n",
      "Clinical Genetics ENTITY\n",
      "North West Genomic Laboratory Hub ENTITY\n",
      "Manchester ENTITY\n",
      "Manchester ENTITY\n",
      "M13 9WL ENTITY\n",
      "Laboratory Report\n",
      " ENTITY\n",
      "Testing\n",
      " ENTITY\n",
      "TESTING ENTITY\n",
      "COPY ENTITY\n",
      "REPORTING REFERRAL INFORMATION TEXT' ENTITY\n",
      "BOX ENTITY\n",
      "DELETE ENTITY\n",
      "clinician ENTITY\n",
      "notes ENTITY\n",
      "Clinical Indication ENTITY\n",
      "RXXX ENTITY\n",
      "Genetic ENTITY\n",
      "Result ENTITY\n",
      "Interpretation\n",
      " ENTITY\n",
      "pathogenic ENTITY\n",
      "variants ENTITY\n",
      "genetic ENTITY\n",
      "individual's ENTITY\n",
      "clinical symptoms ENTITY\n",
      "identified ENTITY\n",
      "diagnosis ENTITY\n",
      "genetic disorder ENTITY\n",
      "DNA abnormalities ENTITY\n",
      "technology\n",
      " ENTITY\n",
      "Appendix I overleaf ENTITY\n",
      "Authorised ENTITY\n",
      "Forename ENTITY\n",
      "Jack Report ENTITY\n",
      "HIM ENTITY\n",
      "DoB ENTITY\n",
      "APPENDIX I ENTITY\n",
      "Test Methodology ENTITY\n",
      "indication ENTITY\n",
      "Thoracic aortic aneurysm ENTITY\n",
      "dissection ENTITY\n",
      "R125.1 ENTITY\n",
      "R125.2)\n",
      " ENTITY\n",
      "Laboratory process ENTITY\n",
      "Chemagic Prime ENTITY\n",
      "samples ENTITY\n",
      "NWGLH\n",
      "Enrichment method ENTITY\n",
      "Agilent SureSelect custom design\n",
      "Sequencing method ENTITY\n",
      "Illumina NextSeq 550 ENTITY\n",
      "Bioinformatics ENTITY\n",
      "quality control ENTITY\n",
      "Congenica ENTITY\n",
      "Alignment ENTITY\n",
      "SNV/Indel calling ENTITY\n",
      "Sentieon DNASeq ENTITY\n",
      "CNV ENTITY\n",
      "calling ENTITY\n",
      "Exome Depth\n",
      "Panel(s ENTITY\n",
      "Thoracic aortic aneurysm ENTITY\n",
      "dissection ENTITY\n",
      "v1.2 ENTITY\n",
      "R125.1 ENTITY\n",
      "R125.2)\n",
      "ABL1 ENTITY\n",
      "ACTA2 ENTITY\n",
      "BGN ENTITY\n",
      "CBS ENTITY\n",
      "COL3A1 ENTITY\n",
      "COL5A1 ENTITY\n",
      "COL5A2 ENTITY\n",
      "EFEMP2 ENTITY\n",
      "ELN ENTITY\n",
      "FBLN5 ENTITY\n",
      "FBN1 ENTITY\n",
      "FBN2 ENTITY\n",
      "FKBP14 ENTITY\n",
      "FLNA ENTITY\n",
      "FOXE3 ENTITY\n",
      "LOX ENTITY\n",
      "MFAP5 ENTITY\n",
      "MYH11 ENTITY\n",
      "MYLK ENTITY\n",
      "NOTCH1 ENTITY\n",
      "PLOD1 ENTITY\n",
      "PRKG1 ENTITY\n",
      "SKI ENTITY\n",
      "SLC2A10 ENTITY\n",
      "SMAD2 ENTITY\n",
      "SMAD3 ENTITY\n",
      "SMAD4 ENTITY\n",
      "SMAD6 ENTITY\n",
      "TGFB2 ENTITY\n",
      "TGFB3 ENTITY\n",
      "TGFBR1 ENTITY\n",
      "TGFBR2 ENTITY\n",
      "panel ENTITY\n",
      "diagnostic grade ENTITY\n",
      "PanelApp panel ENTITY\n",
      "https://nhsgms-panelapp.genomicsengland.co.uk/ ENTITY\n",
      "specific ENTITY\n",
      "version ENTITY\n",
      "test ENTITY\n",
      "detect ENTITY\n",
      "single nucleotide variants ENTITY\n",
      "SNVs ENTITY\n",
      "small insertions ENTITY\n",
      "deletions ENTITY\n",
      "INDELs ENTITY\n",
      "copy number variations ENTITY\n",
      "single exon ENTITY\n",
      "deletions ENTITY\n",
      "duplications ENTITY\n",
      "panel targets protein ENTITY\n",
      "exons ENTITY\n",
      "exon intron boundaries ENTITY\n",
      "non-coding ENTITY\n",
      "deep intronic ENTITY\n",
      "variants ENTITY\n",
      "variants ENTITY\n",
      "patient’s ENTITY\n",
      "phenotype ENTITY\n",
      "result ENTITY\n",
      "phenotype ENTITY\n",
      "patient ENTITY\n",
      "please ENTITY\n",
      "laboratory ENTITY\n",
      "analysis ENTITY\n",
      "Coverage ENTITY\n",
      "coding regions ENTITY\n",
      "diagnostic grade ENTITY\n",
      "panel ENTITY\n",
      "50X ENTITY\n",
      "Patient ENTITY\n",
      "request ENTITY\n",
      "Limitations ENTITY\n",
      "test ENTITY\n",
      "detection ENTITY\n",
      "expansions ENTITY\n",
      "variants ENTITY\n",
      "mitochondrial DNA ENTITY\n",
      "mtDNA ENTITY\n",
      "test ENTITY\n",
      "balanced ENTITY\n",
      "translocations ENTITY\n",
      "inversions ENTITY\n",
      "detect ENTITY\n",
      "low-level ENTITY\n",
      "mosaicism ENTITY\n",
      "inter-locus gene ENTITY\n",
      "conversion ENTITY\n",
      "sensitivity ENTITY\n",
      "detect ENTITY\n",
      "variants ENTITY\n",
      "genes ENTITY\n",
      "gene ENTITY\n",
      "genome ENTITY\n",
      "gene ENTITY\n",
      "suboptimal ENTITY\n",
      "coverage ENTITY\n",
      "regions ENTITY\n",
      "R125 ENTITY\n",
      "CBS ENTITY\n",
      "exons 3-17 (NM_000071.2 ENTITY\n",
      "Please contact ENTITY\n",
      "laboratory\n",
      " ENTITY\n",
      "details ENTITY\n",
      "Confirmation ENTITY\n",
      "findings ENTITY\n",
      "SNV ENTITY\n",
      "INDEL calls ENTITY\n",
      "internally ENTITY\n",
      "quality ENTITY\n",
      "metrics ENTITY\n",
      "Sanger sequencing ENTITY\n",
      "presence ENTITY\n",
      "CNV ENTITY\n",
      "independent method ENTITY\n",
      "MLPA ENTITY\n",
      "custom droplet digital PCR ENTITY\n",
      "QF-PCR assay ENTITY\n",
      "laboratory-developed test ENTITY\n",
      "locally validated ENTITY\n",
      "sensitivity ENTITY\n",
      "methodology ENTITY\n",
      "SNVs ENTITY\n",
      "CI ENTITY\n",
      "INDELs ENTITY\n",
      "CI ENTITY\n",
      "CNVs ENTITY\n",
      "CI ENTITY\n",
      "specificity ENTITY\n",
      "variant ENTITY\n",
      "types ENTITY\n",
      "diagnosis ENTITY\n",
      "genetic disorder ENTITY\n",
      "DNA abnormalities ENTITY\n",
      "technology ENTITY\n",
      "accreditations ENTITY\n",
      "test ENTITY\n",
      "UKAS ENTITY\n",
      "accredited ENTITY\n",
      "test ENTITY\n",
      "locally ENTITY\n",
      "UKAS ENTITY\n",
      "assessment ENTITY\n",
      "laboratory ENTITY\n",
      "details ENTITY\n",
      "Page 2 ENTITY\n"
     ]
    }
   ],
   "execution_count": 24
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-02-10T07:22:54.759414Z",
     "start_time": "2026-02-10T07:22:54.680845Z"
    }
   },
   "cell_type": "code",
   "source": "displacy.render(doc, jupyter=False, style='ent')",
   "id": "37aa5a17959e9d4b",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'<div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    North West Genomic Laboratory Hub\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n (\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Manchester Site\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n)<br>Manchester Centre for \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Genomic Medicine\\n6th Floor\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    St Mary\\'s Hospital\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Manchester\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    M13 9WL\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n<br>Scientific Operational Director: Dr \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    E. Howard\\nhttps://mft.nhs.uk/nwglh/ mft.genomics@nhs.net Tel +44(0\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n) 161 276 6122<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Forename\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Jack\\nSurname\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    HIM\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Sex\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n:<br>Report No: R25-00EK-3<br>Hospital No: Not provided<br>Other No: Not provided<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    DoB\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: 18/02/1987 \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Sample\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Blood\\nNHS No\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n:<br>Hive Order \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    ID\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: 1000152866 Path. No: \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    ACC1810A1\\nCollected\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: Not provided \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Activated\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: 14/10/2025<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Reported\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: 14/10/2025<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Referred\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n by: Jonathan Edgerley, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Clinical Genetics\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    North West Genomic Laboratory Hub\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n (\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Manchester\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n), \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Manchester\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    M13 9WL\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, (jonathan.<br>edgerley@mft.nhs.uk)<br>Genomics \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Laboratory Report\\n\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\nReason for \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Testing\\n\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n[REASON FOR \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    TESTING\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n - \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    COPY\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n BELOW TEXT TO \\'\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    REPORTING REFERRAL INFORMATION TEXT\\'\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    BOX\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n THEN \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    DELETE\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n]<br>[Insert referring \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    clinician\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    notes\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n]<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Clinical Indication\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: [Insert Clinical Indication Name (\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    RXXX\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n)].<br>RESULT SUMMARY:<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Genetic\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n cause not identified.<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Result\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n and \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Interpretation\\n\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\nNo \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    pathogenic\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    variants\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n that explain the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    genetic\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n cause of this \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    individual\\'s\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    clinical symptoms\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n were \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    identified\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n.<br>This result does not rule out the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    diagnosis\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n of a \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    genetic disorder\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n since some \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    DNA abnormalities\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n are undetectable by the applied \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    technology\\n\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n(see \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Appendix I overleaf\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n).<br>Reported By:<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Authorised\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n By:<br>Jonathan Edgerley<br>Clinical Scientist<br>Jonathan Edgerley<br>Clinical Scientist<br>Page 1 of 2<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Forename\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Jack Report\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n No: R25-00EK-3<br>Surname: \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    HIM\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    DoB\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: 18/02/1987<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    APPENDIX I\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n - \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Test Methodology\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n:<br>Clinical \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    indication\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Thoracic aortic aneurysm\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n and \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    dissection\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n (\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    R125.1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    R125.2)\\n\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Laboratory process\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n<br>DNA extraction: \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Chemagic Prime\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n for \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    samples\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n extracted by \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    NWGLH\\nEnrichment method\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Agilent SureSelect custom design\\nSequencing method\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Illumina NextSeq 550\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Bioinformatics\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n and \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    quality control\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Congenica\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n – \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Alignment\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n and \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    SNV/Indel calling\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n – \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Sentieon DNASeq\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n; \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    CNV\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    calling\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n – \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Exome Depth\\nPanel(s\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n) applied: \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Thoracic aortic aneurysm\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n and \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    dissection\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    v1.2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n (\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    R125.1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    R125.2)\\nABL1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    ACTA2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    BGN\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    CBS\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    COL3A1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    COL5A1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    COL5A2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    EFEMP2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    ELN\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    FBLN5\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    FBN1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    FBN2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    FKBP14\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    FLNA\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    FOXE3\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    LOX\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    MFAP5\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    MYH11\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    MYLK\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    NOTCH1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    PLOD1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n,<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    PRKG1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    SKI\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    SLC2A10\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    SMAD2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    SMAD3\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    SMAD4\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    SMAD6\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    TGFB2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    TGFB3\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    TGFBR1\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    TGFBR2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n.<br>The \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    panel\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n includes \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    diagnostic grade\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n (green) genes from the associated \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    PanelApp panel\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n (\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    https://nhsgms-panelapp.genomicsengland.co.uk/\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n) for the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    specific\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    version\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n detailed<br>above. This \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    test\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n is used to \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    detect\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    single nucleotide variants\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n (\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    SNVs\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n), \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    small insertions\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n and \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    deletions\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n (\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    INDELs\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n) and \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    copy number variations\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n (defined as \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    single exon\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n or larger<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    deletions\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n and \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    duplications\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n). This \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    panel targets protein\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n coding \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    exons\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    exon intron boundaries\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n (± 8 bps) and selected \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    non-coding\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    deep intronic\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    variants\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n. Only \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    variants\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n relevant<br>to this \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    patient’s\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n submitted \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    phenotype\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n have been reported. If the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    result\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n does not fully account for the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    phenotype\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n observed in the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    patient\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    please\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n contact the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    laboratory\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n to<br>discuss if further \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    analysis\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n may be appropriate.<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Coverage\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: &gt;97% of the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    coding regions\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n of the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    diagnostic grade\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n (green) genes in the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    panel\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n above are covered to a minimum depth of \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    50X\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n. \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Patient\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n specific coverage for<br>individual genes can be provided on \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    request\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n.<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Limitations\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: This \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    test\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n cannot be used for the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    detection\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n of repeat \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    expansions\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n or \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    variants\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n in \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    mitochondrial DNA\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n (\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    mtDNA\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n). The \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    test\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n does not identify \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    balanced\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    translocations\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n or<br>complex \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    inversions\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, and it may not \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    detect\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    low-level\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    mosaicism\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n or \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    inter-locus gene\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    conversion\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n. The \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    sensitivity\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n to \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    detect\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    variants\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n may be limited in \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    genes\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n where some, or all, of<br>the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    gene\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n is duplicated in the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    genome\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n or the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    gene\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n has \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    suboptimal\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    coverage\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n. Specific \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    regions\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n for \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    R125\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n include \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    CBS\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    exons 3-17 (NM_000071.2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n). \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Please contact\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    laboratory\\n\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\nfor additional \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    details\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n.<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Confirmation\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n of \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    findings\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    SNV\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n and \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    INDEL calls\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n which do not meet \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    internally\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n validated \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    quality\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    metrics\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n are confirmed by \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Sanger sequencing\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n. The \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    presence\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, but not the<br>extent, of \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    CNV\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n calls is confirmed by an \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    independent method\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n (\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    MLPA\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    custom droplet digital PCR\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n assay, or custom \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    QF-PCR assay\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n).<br>Analytical validation: This \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    laboratory-developed test\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n has been \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    locally validated\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n. The \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    sensitivity\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n of this \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    methodology\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n for \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    SNVs\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n is 100% (95% \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    CI\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: 99.78% to 100%), \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    INDELs\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n is<br>95.86% (95% \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    CI\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: 91.65% to 98.32%) and \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    CNVs\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n is 100% (95% \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    CI\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: 93.28% to 100%), and \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    specificity\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n is &gt;99.9% for most \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    variant\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    types\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n. A normal result does not rule out the<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    diagnosis\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n of a \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    genetic disorder\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n since some \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    DNA abnormalities\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n are undetectable by the applied \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    technology\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n.<br>Regulation and \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    accreditations\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n: This \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    test\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n is not \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    UKAS\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    accredited\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n. This \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    test\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n has been validated \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    locally\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n and will be submitted for \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    UKAS\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    assessment\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n imminently.<br>Please contact the \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    laboratory\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n for additional \\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    details\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n, if required.<br>\\n<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\\n    Page 2\\n    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">ENTITY</span>\\n</mark>\\n of 2</div>'"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 43
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-02-10T06:43:39.982423Z",
     "start_time": "2026-02-10T06:41:59.555478Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import spacy\n",
    "import scispacy\n",
    "\n",
    "from scispacy.linking import EntityLinker\n",
    "\n",
    "nlp = spacy.load(\"en_core_sci_sm\")\n",
    "\n",
    "# This line takes a while, because we have to download ~1GB of data\n",
    "# and load a large JSON file (the knowledge base). Be patient!\n",
    "# Thankfully it should be faster after the first time you use it, because\n",
    "# the downloads are cached.\n",
    "# NOTE: The resolve_abbreviations parameter is optional, and requires that\n",
    "# the AbbreviationDetector pipe has already been added to the pipeline. Adding\n",
    "# the AbbreviationDetector pipe and setting resolve_abbreviations to True means\n",
    "# that linking will only be performed on the long form of abbreviations.\n",
    "nlp.add_pipe(\"scispacy_linker\", config={\"resolve_abbreviations\": True, \"linker_name\": \"umls\"})\n",
    "\n",
    "doc = nlp(clean_text)\n",
    "\n",
    "# Let's look at a random entity!\n",
    "entity = doc.ents[1]\n",
    "\n",
    "print(\"Name: \", entity)\n",
    "\n",
    "# Each entity is linked to UMLS with a score\n",
    "# (currently just char-3gram matching).\n",
    "linker = nlp.get_pipe(\"scispacy_linker\")\n",
    "for umls_ent in entity._.kb_ents:\n",
    "    print(linker.kb.cui_to_entity[umls_ent[0]])"
   ],
   "id": "3e6007cc2e52eca7",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name:  Manchester Site\n"
     ]
    }
   ],
   "execution_count": 27
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
