**North West Genomic Laboratory Hub (Manchester Site)**
Manchester Centre for Genomic Medicine
6th Floor, St Mary's Hospital, Manchester, M13 9WL
Scientific Operational Director: Dr E. Howard
https://mft.nhs.uk/nwglh/ mft.genomics@nhs.net Tel +44(0) 161 276 6122



Forename: **Jack**

Surname: **HIM**
Sex:



Report No: **R25-00EK-3**
Hospital No: **Not provided**
Other No: **Not provided**



DoB: **18/02/1987** Sample: **Blood**

NHS No:

Hive Order ID: **1000152866** Path. No: **ACC1810A1**


Collected: **Not provided** Activated: **14/10/2025**
Reported: **14/10/2025**


Referred by: Jonathan Edgerley, Clinical Genetics, North West Genomic Laboratory Hub (Manchester), Manchester, M13 9WL, (jonathan.
edgerley@mft.nhs.uk)

# **Genomics Laboratory Report**


**Reason for Testing**

[REASON FOR TESTING - COPY BELOW TEXT TO 'REPORTING REFERRAL INFORMATION TEXT' BOX THEN DELETE]

[Insert referring clinician notes]
**Clinical Indication:** [Insert Clinical Indication Name (RXXX)].


**RESULT SUMMARY:**


Genetic cause not identified.


**Result and Interpretation**

No pathogenic variants that explain the genetic cause of this individual's clinical symptoms were identified.


This result does not rule out the diagnosis of a genetic disorder since some DNA abnormalities are undetectable by the applied technology
(see Appendix I overleaf).


Reported By:
Authorised By:
Jonathan Edgerley
Clinical Scientist


Jonathan Edgerley
Clinical Scientist


Page 1 of 2


Forename: **Jack** Report No: **R25-00EK-3**

Surname: **HIM** DoB: **18/02/1987**


_**APPENDIX I - Test Methodology:**_


_**Clinical indication:**_ _Thoracic aortic aneurysm and dissection (R125.1, R125.2)_


_**Laboratory process**_
_**DNA extraction:**_ _Chemagic Prime for samples extracted by NWGLH_
_**Enrichment method:**_ _Agilent SureSelect custom design_
_**Sequencing method:**_ _Illumina NextSeq 550_
_**Bioinformatics and quality control:**_ _Congenica – Alignment and SNV/Indel calling – Sentieon DNASeq; CNV calling – Exome Depth_
_**Panel(s) applied:**_ _Thoracic aortic aneurysm and dissection v1.2 (R125.1, R125.2)_
_ABL1, ACTA2, BGN, CBS, COL3A1, COL5A1, COL5A2, EFEMP2, ELN, FBLN5, FBN1, FBN2, FKBP14, FLNA, FOXE3, LOX, MFAP5, MYH11, MYLK, NOTCH1, PLOD1,_
_PRKG1, SKI, SLC2A10, SMAD2, SMAD3, SMAD4, SMAD6, TGFB2, TGFB3, TGFBR1, TGFBR2._


_The panel includes diagnostic grade (green) genes from the associated PanelApp panel (https://nhsgms-panelapp.genomicsengland.co.uk/) for the specific version detailed_
_above. This test is used to detect single nucleotide variants (SNVs), small insertions and deletions (INDELs) and copy number variations (defined as single exon or larger_
_deletions and duplications). This panel targets protein coding exons, exon intron boundaries (± 8 bps) and selected non-coding, deep intronic variants. Only variants relevant_
_to this patient’s submitted phenotype have been reported. If the result does not fully account for the phenotype observed in the patient, please contact the laboratory to_
_discuss if further analysis may be appropriate._


_**Coverage:**_ _>97% of the coding regions of the diagnostic grade (green) genes in the panel above are covered to a minimum depth of 50X. Patient specific coverage for_
_individual genes can be provided on request._


_**Limitations:**_ _This test cannot be used for the detection of repeat expansions or variants in mitochondrial DNA (mtDNA). The test does not identify balanced translocations or_
_complex inversions, and it may not detect low-level mosaicism or inter-locus gene conversion. The sensitivity to detect variants may be limited in genes where some, or all, of_
_the gene is duplicated in the genome or the gene has suboptimal coverage. Specific regions for R125 include CBS exons 3-17 (NM_000071.2). Please contact the laboratory_
_for additional details._


_**Confirmation of findings:**_ _SNV and INDEL calls which do not meet internally validated quality metrics are confirmed by Sanger sequencing. The presence, but not the_
_extent, of CNV calls is confirmed by an independent method (MLPA, custom droplet digital PCR assay, or custom QF-PCR assay)._


_**Analytical validation:**_ _This laboratory-developed test has been locally validated. The sensitivity of this methodology for SNVs is 100% (95% CI: 99.78% to 100%), INDELs is_
_95.86% (95% CI: 91.65% to 98.32%) and CNVs is 100% (95% CI: 93.28% to 100%), and specificity is >99.9% for most variant types. A normal result does not rule out the_
_diagnosis of a genetic disorder since some DNA abnormalities are undetectable by the applied technology._


_**Regulation and accreditations:**_ _This test is not UKAS accredited. This test has been validated locally and will be submitted for UKAS assessment imminently._


_Please contact the laboratory for additional details, if required._


Page 2 of 2


