**North West Genomic Laboratory Hub (Manchester Site)**
Manchester Centre for Genomic Medicine
6th Floor, St Mary's Hospital, Manchester, M13 9WL
Scientific Operational Director: Dr E. Howard
https://mft.nhs.uk/nwglh/ mft.genomics@nhs.net Tel +44(0) 161 276 6122



Forename: **Aphrodite**

Surname: **DLIMStest**

Sex: **Female**



Report No: **R25-00F1-1**
Hospital No: **Not provided**
Other No: **Not provided**



DoB: Sample: **Blood**

NHS No:

Path. No: **ACC1914A1**


Collected: **Not provided** Received: **13/10/2025 13:44** Activated: **13/10/2025**
Reported: **06/11/2025**


Referred by: Invalid Email, Connolly Hospital, Dublin, D15 X40D, (thisisnotanemailaddress2025@mft.com)

# **Genomics Laboratory Report**


**Reason for Testing**


**RESULT SUMMARY:**


Pathogenic variant detected.
Confirms a genetic diagnosis of autosomal dominant XXXX OR GENE-related disease.


**Result and Interpretation**

This individual is heterozygous for a pathogenic GENE variant, VARIANT (details in Appendix II overleaf). Monoallelic pathogenic GENE
variants cause (a range of disorders including) autosomal dominant XXXX (OMIM# XXXX).


This result confirms a genetic diagnosis of autosomal dominant XXXX OR GENE-related disease.


[ONLY INCLUDE AS APPROPRIATE]: Testing of this individual’s parents is available (via referral to a clinical genetics service) to
investigate if the variant has arisen _de novo_, and to provide advice to their parents regarding future pregnancies, if appropriate.
OR
This result has implications for other family members. Testing for this variant is available to other relatives of this individual, as appropriate
(via referral to a clinical genetics service).


[REASON FOR TESTING - COPY BELOW TEXT TO 'REPORTING REFERRAL INFORMATION TEXT' BOX THEN DELETE]

[Insert referring clinician notes]
**Clinical Indication:** [Insert Clinical Indication Name (RXXX)].


Reported By:
Authorised By:
Fraser Suttie
Administrator


Fraser Suttie
Administrator


Page 1 of 2


Forename: **Aphrodite** Report No: **R25-00F1-1**

Surname: **DLIMStest** DoB:


_**APPENDIX I - Test Methodology:**_


_**Clinical indication:**_ _Factor II deficiency (R112.1)_


_**Laboratory process**_
_**DNA extraction:**_ _Chemagic Prime for samples extracted by NWGLH_
_**Enrichment method:**_ _Agilent SureSelect custom design_
_**Sequencing method:**_ _Illumina NextSeq 550_
_**Bioinformatics and quality control:**_ _Congenica – Alignment and SNV/Indel calling – Sentieon DNASeq; CNV calling – Exome Depth_
_**Panel(s) applied:**_ _Factor II deficiency (R112.1)_


_The panel includes diagnostic grade (green) genes from the associated PanelApp panel (https://nhsgms-panelapp.genomicsengland.co.uk/) for the specific version detailed_
_above. This test is used to detect single nucleotide variants (SNVs), small insertions and deletions (INDELs) and copy number variations (defined as single exon or larger_
_deletions and duplications). This panel targets protein coding exons, exon intron boundaries (± 8 bps) and selected non-coding, deep intronic variants. Only variants relevant_
_to this patient’s submitted phenotype have been reported. If the result does not fully account for the phenotype observed in the patient, please contact the laboratory to_
_discuss if further analysis may be appropriate._


_**Coverage:**_ _>97% of the coding regions of the diagnostic grade (green) genes in the panel above are covered to a minimum depth of 50X. Patient specific coverage for_
_individual genes can be provided on request._


_**Limitations:**_ _This test cannot be used for the detection of repeat expansions or variants in mitochondrial DNA (mtDNA). The test does not identify balanced translocations or_
_complex inversions, and it may not detect low-level mosaicism or inter-locus gene conversion. The sensitivity to detect variants may be limited in genes where some, or all, of_
_the gene is duplicated in the genome or the gene has suboptimal coverage. Please contact the laboratory for additional details._


_**Confirmation of findings:**_ _SNV and INDEL calls which do not meet internally validated quality metrics are confirmed by Sanger sequencing. The presence, but not the_
_extent, of CNV calls is confirmed by an independent method (MLPA, custom droplet digital PCR assay, or custom QF-PCR assay)._


_**Analytical validation:**_ _This laboratory-developed test has been locally validated. The sensitivity of this methodology for SNVs is 100% (95% CI: 99.78% to 100%), INDELs is_
_95.86% (95% CI: 91.65% to 98.32%) and CNVs is 100% (95% CI: 93.28% to 100%), and specificity is >99.9% for most variant types. A normal result does not rule out the_
_diagnosis of a genetic disorder since some DNA abnormalities are undetectable by the applied technology._


_**Regulation and accreditations:**_ _This test is not UKAS accredited. This test has been validated locally and will be submitted for UKAS assessment imminently._


_Please contact the laboratory for additional details, if required._


Page 2 of 2


